Should be bad parsing instead of bad compile

``` r
library(rxode2)
#> rxode2 3.0.2.9000 using 8 threads (see ?getRxThreads)
#>   no cache: create with `rxCreateCache()`

mod2 <- function() {
  ini({
    ## Table 3 from Savic 2007
    cl  <- 17.2 # (L/hr)
    vc  <- 45.1 # L
    ka  <- 0.38 # 1/hr
    mtt <- 1.37 # hr

    f2 <-0.5    # Fraction of 1st Order portion
    n   <- 20.1
  })
  model({
    k           <- cl/vc
    bio <- 1-f2
    ktr = (n+1)/mtt
    ## note that lgammafn is the same as lgamma in R.
    d/dt(depot1) = exp(log(bio*podo(depot))+
                         log(ktr)+n*log(ktr*tad(depot))-
                         ktr*tad(depot)-lgammafn(n+1))-ka*depot1
    d/dt(depot2) <- -ka*depot2
    f(depot2) <-f2
    d/dt(cen)   <- ka*depot1 + ka*depot2-k*cen
  })
}

mod2 <- mod2()

mod2$simulationModel
#> using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’
#> In file included from rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:149:
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c: In function ‘rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt’:
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:99: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                                   ^~~~~~~
#> /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:22:33: note: in definition of macro ‘_safe_log’
#>    22 | #define _safe_log(a) _safe_log_(a, _solveData)
#>       |                                 ^
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:92: note: in expansion of macro ‘_podo1’
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                            ^~~~~~
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:99: note: each undeclared identifier is reported only once for each function it appears in
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                                   ^~~~~~~
#> /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:22:33: note: in definition of macro ‘_safe_log’
#>    22 | #define _safe_log(a) _safe_log_(a, _solveData)
#>       |                                 ^
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:92: note: in expansion of macro ‘_podo1’
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                            ^~~~~~
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c: In function ‘rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_lhs’:
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:715:91: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   715 |   __DDtStateVar_0__ = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                           ^~~~~~~
#> /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:22:33: note: in definition of macro ‘_safe_log’
#>    22 | #define _safe_log(a) _safe_log_(a, _solveData)
#>       |                                 ^
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:715:84: note: in expansion of macro ‘_podo1’
#>   715 |   __DDtStateVar_0__ = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                    ^~~~~~
#> make: *** [/usr/lib/R/etc/Makeconf:195: rx_c1b42db8d18f7ac7822d3b8d3093b99f_.o] Error 1
#> error building model
#> ccache gcc -I"/usr/share/R/include" -DNDEBUG      -O3  -I"/home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include" -I"/home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include" -fpic  -O0 -Wall -c rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c -o rx_c1b42db8d18f7ac7822d3b8d3093b99f_.o
#> using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’
#> In file included from rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:149:
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c: In function ‘rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt’:
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:99: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                                   ^~~~~~~
#> /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:22:33: note: in definition of macro ‘_safe_log’
#>    22 | #define _safe_log(a) _safe_log_(a, _solveData)
#>       |                                 ^
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:92: note: in expansion of macro ‘_podo1’
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                            ^~~~~~
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:99: note: each undeclared identifier is reported only once for each function it appears in
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                                   ^~~~~~~
#> /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:22:33: note: in definition of macro ‘_safe_log’
#>    22 | #define _safe_log(a) _safe_log_(a, _solveData)
#>       |                                 ^
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:633:92: note: in expansion of macro ‘_podo1’
#>   633 |   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                            ^~~~~~
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c: In function ‘rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_lhs’:
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:715:91: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   715 |   __DDtStateVar_0__ = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                           ^~~~~~~
#> /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:22:33: note: in definition of macro ‘_safe_log’
#>    22 | #define _safe_log(a) _safe_log_(a, _solveData)
#>       |                                 ^
#> rx_c1b42db8d18f7ac7822d3b8d3093b99f_.c:715:84: note: in expansion of macro ‘_podo1’
#>   715 |   __DDtStateVar_0__ = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>       |                                                                                    ^~~~~~
#> make: *** [/usr/lib/R/etc/Makeconf:195: rx_c1b42db8d18f7ac7822d3b8d3093b99f_.o] Error 1
#> // Define translation state order for 3 states
#> #define __DDT0__ 0 // depot1
#> #define __DDT1__ 1 // depot2
#> #define __DDT2__ 2 // cen
#> #define _getRxSolve_ _rx69b11d637ca1f0621093f3eab6d8a8b00
#> #define _evalUdf _rx69b11d637ca1f0621093f3eab6d8a8b01
#> #define _solveData _rx69b11d637ca1f0621093f3eab6d8a8b02
#> #define _assign_ptr _rx69b11d637ca1f0621093f3eab6d8a8b03
#> #define _rxRmModelLib _rx69b11d637ca1f0621093f3eab6d8a8b04
#> #define _rxGetModelLib _rx69b11d637ca1f0621093f3eab6d8a8b05
#> #define _old_c _rx69b11d637ca1f0621093f3eab6d8a8b06
#> #define _ptrid _rx69b11d637ca1f0621093f3eab6d8a8b07
#> #define _rxIsCurrentC _rx69b11d637ca1f0621093f3eab6d8a8b08
#> #define _sumPS _rx69b11d637ca1f0621093f3eab6d8a8b09
#> #define _prodPS _rx69b11d637ca1f0621093f3eab6d8a8b010
#> #define _prodType _rx69b11d637ca1f0621093f3eab6d8a8b011
#> #define _sumType _rx69b11d637ca1f0621093f3eab6d8a8b012
#> #define _update_par_ptr _rx69b11d637ca1f0621093f3eab6d8a8b013
#> #define _getParCov _rx69b11d637ca1f0621093f3eab6d8a8b014
#> #define _rxode2_rxAssignPtr _rx69b11d637ca1f0621093f3eab6d8a8b015
#> #define _rxQr _rx69b11d637ca1f0621093f3eab6d8a8b016
#> #define _compareFactorVal _rx69b11d637ca1f0621093f3eab6d8a8b017
#> #define _sum _rx69b11d637ca1f0621093f3eab6d8a8b018
#> #define _udf _rx69b11d637ca1f0621093f3eab6d8a8b019
#> #define _sign _rx69b11d637ca1f0621093f3eab6d8a8b020
#> #define _prod _rx69b11d637ca1f0621093f3eab6d8a8b021
#> #define _max _rx69b11d637ca1f0621093f3eab6d8a8b022
#> #define _min _rx69b11d637ca1f0621093f3eab6d8a8b023
#> #define _transit4P _rx69b11d637ca1f0621093f3eab6d8a8b024
#> #define _transit3P _rx69b11d637ca1f0621093f3eab6d8a8b025
#> #define _assignFuns0 _rx69b11d637ca1f0621093f3eab6d8a8b026
#> #define _assignFuns _rx69b11d637ca1f0621093f3eab6d8a8b027
#> #define _rxord _rx69b11d637ca1f0621093f3eab6d8a8b028
#> #define __assignFuns2 _rx69b11d637ca1f0621093f3eab6d8a8b029
#> #define _llikCauchyDscale _rx69b11d637ca1f0621093f3eab6d8a8b030
#> #define _llikCauchyDlocation _rx69b11d637ca1f0621093f3eab6d8a8b031
#> #define _llikCauchy _rx69b11d637ca1f0621093f3eab6d8a8b032
#> #define _llikGammaDrate _rx69b11d637ca1f0621093f3eab6d8a8b033
#> #define _llikGammaDshape _rx69b11d637ca1f0621093f3eab6d8a8b034
#> #define _llikGamma _rx69b11d637ca1f0621093f3eab6d8a8b035
#> #define _llikWeibullDscale _rx69b11d637ca1f0621093f3eab6d8a8b036
#> #define _llikWeibullDshape _rx69b11d637ca1f0621093f3eab6d8a8b037
#> #define _llikWeibull _rx69b11d637ca1f0621093f3eab6d8a8b038
#> #define _llikUnifDbeta _rx69b11d637ca1f0621093f3eab6d8a8b039
#> #define _llikUnifDalpha _rx69b11d637ca1f0621093f3eab6d8a8b040
#> #define _llikUnif _rx69b11d637ca1f0621093f3eab6d8a8b041
#> #define _llikGeomDp _rx69b11d637ca1f0621093f3eab6d8a8b042
#> #define _llikGeom _rx69b11d637ca1f0621093f3eab6d8a8b043
#> #define _llikFDdf2 _rx69b11d637ca1f0621093f3eab6d8a8b044
#> #define _llikFDdf1 _rx69b11d637ca1f0621093f3eab6d8a8b045
#> #define _llikF _rx69b11d637ca1f0621093f3eab6d8a8b046
#> #define _llikExpDrate _rx69b11d637ca1f0621093f3eab6d8a8b047
#> #define _llikExp _rx69b11d637ca1f0621093f3eab6d8a8b048
#> #define _llikChisqDdf _rx69b11d637ca1f0621093f3eab6d8a8b049
#> #define _llikChisq _rx69b11d637ca1f0621093f3eab6d8a8b050
#> #define _llikTDsd _rx69b11d637ca1f0621093f3eab6d8a8b051
#> #define _llikTDmean _rx69b11d637ca1f0621093f3eab6d8a8b052
#> #define _llikTDdf _rx69b11d637ca1f0621093f3eab6d8a8b053
#> #define _llikT _rx69b11d637ca1f0621093f3eab6d8a8b054
#> #define _llikBetaDshape2 _rx69b11d637ca1f0621093f3eab6d8a8b055
#> #define _llikBetaDshape1 _rx69b11d637ca1f0621093f3eab6d8a8b056
#> #define _llikBeta _rx69b11d637ca1f0621093f3eab6d8a8b057
#> #define _llikNbinomMuDmu _rx69b11d637ca1f0621093f3eab6d8a8b058
#> #define _llikNbinomMu _rx69b11d637ca1f0621093f3eab6d8a8b059
#> #define _llikNbinomDprob _rx69b11d637ca1f0621093f3eab6d8a8b060
#> #define _llikNbinom _rx69b11d637ca1f0621093f3eab6d8a8b061
#> #define _llikBinomDprob _rx69b11d637ca1f0621093f3eab6d8a8b062
#> #define _llikBinom _rx69b11d637ca1f0621093f3eab6d8a8b063
#> #define _llikPoisDlambda _rx69b11d637ca1f0621093f3eab6d8a8b064
#> #define _llikPois _rx69b11d637ca1f0621093f3eab6d8a8b065
#> #define _llikNormDsd _rx69b11d637ca1f0621093f3eab6d8a8b066
#> #define _llikNormDmean _rx69b11d637ca1f0621093f3eab6d8a8b067
#> #define _llikNorm _rx69b11d637ca1f0621093f3eab6d8a8b068
#> #define simeps _rx69b11d637ca1f0621093f3eab6d8a8b069
#> #define simeta _rx69b11d637ca1f0621093f3eab6d8a8b070
#> #define expit _rx69b11d637ca1f0621093f3eab6d8a8b071
#> #define logit _rx69b11d637ca1f0621093f3eab6d8a8b072
#> #define gammapDer _rx69b11d637ca1f0621093f3eab6d8a8b073
#> #define lowergamma _rx69b11d637ca1f0621093f3eab6d8a8b074
#> #define uppergamma _rx69b11d637ca1f0621093f3eab6d8a8b075
#> #define gammaqInva _rx69b11d637ca1f0621093f3eab6d8a8b076
#> #define gammaqInv _rx69b11d637ca1f0621093f3eab6d8a8b077
#> #define gammapInva _rx69b11d637ca1f0621093f3eab6d8a8b078
#> #define gammapInv _rx69b11d637ca1f0621093f3eab6d8a8b079
#> #define gammaq _rx69b11d637ca1f0621093f3eab6d8a8b080
#> #define gammap _rx69b11d637ca1f0621093f3eab6d8a8b081
#> #define phi _rx69b11d637ca1f0621093f3eab6d8a8b082
#> #define d2ELUa _rx69b11d637ca1f0621093f3eab6d8a8b083
#> #define dELUa _rx69b11d637ca1f0621093f3eab6d8a8b084
#> #define d2aELU _rx69b11d637ca1f0621093f3eab6d8a8b085
#> #define d2ELU _rx69b11d637ca1f0621093f3eab6d8a8b086
#> #define dELU _rx69b11d637ca1f0621093f3eab6d8a8b087
#> #define ELU _rx69b11d637ca1f0621093f3eab6d8a8b088
#> #define dPReLUa1 _rx69b11d637ca1f0621093f3eab6d8a8b089
#> #define dPReLUa _rx69b11d637ca1f0621093f3eab6d8a8b090
#> #define dPReLU _rx69b11d637ca1f0621093f3eab6d8a8b091
#> #define PReLU _rx69b11d637ca1f0621093f3eab6d8a8b092
#> #define dSwish _rx69b11d637ca1f0621093f3eab6d8a8b093
#> #define Swish _rx69b11d637ca1f0621093f3eab6d8a8b094
#> #define dlReLU _rx69b11d637ca1f0621093f3eab6d8a8b095
#> #define lReLU _rx69b11d637ca1f0621093f3eab6d8a8b096
#> #define dSELU _rx69b11d637ca1f0621093f3eab6d8a8b097
#> #define SELU _rx69b11d637ca1f0621093f3eab6d8a8b098
#> #define d4softplus _rx69b11d637ca1f0621093f3eab6d8a8b099
#> #define d3softplus _rx69b11d637ca1f0621093f3eab6d8a8b0100
#> #define d2softplus _rx69b11d637ca1f0621093f3eab6d8a8b0101
#> #define dsoftplus _rx69b11d637ca1f0621093f3eab6d8a8b0102
#> #define softplus _rx69b11d637ca1f0621093f3eab6d8a8b0103
#> #define d4GELU _rx69b11d637ca1f0621093f3eab6d8a8b0104
#> #define d3GELU _rx69b11d637ca1f0621093f3eab6d8a8b0105
#> #define d2GELU _rx69b11d637ca1f0621093f3eab6d8a8b0106
#> #define dGELU _rx69b11d637ca1f0621093f3eab6d8a8b0107
#> #define GELU _rx69b11d637ca1f0621093f3eab6d8a8b0108
#> #define dReLU _rx69b11d637ca1f0621093f3eab6d8a8b0109
#> #define ReLU _rx69b11d637ca1f0621093f3eab6d8a8b0110
#> #define riweibull _rx69b11d637ca1f0621093f3eab6d8a8b0111
#> #define riunif _rx69b11d637ca1f0621093f3eab6d8a8b0112
#> #define rit_ _rx69b11d637ca1f0621093f3eab6d8a8b0113
#> #define ripois _rx69b11d637ca1f0621093f3eab6d8a8b0114
#> #define ribeta _rx69b11d637ca1f0621093f3eab6d8a8b0115
#> #define rigamma _rx69b11d637ca1f0621093f3eab6d8a8b0116
#> #define rigeom _rx69b11d637ca1f0621093f3eab6d8a8b0117
#> #define rif _rx69b11d637ca1f0621093f3eab6d8a8b0118
#> #define riexp _rx69b11d637ca1f0621093f3eab6d8a8b0119
#> #define richisq _rx69b11d637ca1f0621093f3eab6d8a8b0120
#> #define ricauchy _rx69b11d637ca1f0621093f3eab6d8a8b0121
#> #define rinbinomMu _rx69b11d637ca1f0621093f3eab6d8a8b0122
#> #define rinbinom _rx69b11d637ca1f0621093f3eab6d8a8b0123
#> #define ribinom _rx69b11d637ca1f0621093f3eab6d8a8b0124
#> #define rinorm _rx69b11d637ca1f0621093f3eab6d8a8b0125
#> #define rxweibull _rx69b11d637ca1f0621093f3eab6d8a8b0126
#> #define rxunif _rx69b11d637ca1f0621093f3eab6d8a8b0127
#> #define rxt_ _rx69b11d637ca1f0621093f3eab6d8a8b0128
#> #define rxpois _rx69b11d637ca1f0621093f3eab6d8a8b0129
#> #define rxbeta _rx69b11d637ca1f0621093f3eab6d8a8b0130
#> #define rxgamma _rx69b11d637ca1f0621093f3eab6d8a8b0131
#> #define rxgeom _rx69b11d637ca1f0621093f3eab6d8a8b0132
#> #define rxf _rx69b11d637ca1f0621093f3eab6d8a8b0133
#> #define rxexp _rx69b11d637ca1f0621093f3eab6d8a8b0134
#> #define rxchisq _rx69b11d637ca1f0621093f3eab6d8a8b0135
#> #define rxcauchy _rx69b11d637ca1f0621093f3eab6d8a8b0136
#> #define rxnbinomMu _rx69b11d637ca1f0621093f3eab6d8a8b0137
#> #define rxnbinom _rx69b11d637ca1f0621093f3eab6d8a8b0138
#> #define rxbinom _rx69b11d637ca1f0621093f3eab6d8a8b0139
#> #define rxnorm _rx69b11d637ca1f0621093f3eab6d8a8b0140
#> #define linCmtC _rx69b11d637ca1f0621093f3eab6d8a8b0141
#> #define linCmtB _rx69b11d637ca1f0621093f3eab6d8a8b0142
#> #define linCmtA _rx69b11d637ca1f0621093f3eab6d8a8b0143
#> #include <rxode2_model_shared.h>
#> #define __MAX_PROD__ 0
#> #define _CMT CMT
#> #define _SYNC_simeps_ for (int _svari=_solveData->neps; _svari--;){  if (_solveData->svar[_svari] == 0) {cl = _PP[0];};   if (_solveData->svar[_svari] == 1) {vc = _PP[1];};   if (_solveData->svar[_svari] == 2) {ka = _PP[2];};   if (_solveData->svar[_svari] == 3) {mtt = _PP[3];};   if (_solveData->svar[_svari] == 4) {f2 = _PP[4];};   if (_solveData->svar[_svari] == 5) {n = _PP[5];}; }
#> #define _SYNC_simeta_ for (int _ovari=_solveData->neta; _ovari--;){  if (_solveData->ovar[_ovari] == 0) {cl = _PP[0];};   if (_solveData->ovar[_ovari] == 1) {vc = _PP[1];};   if (_solveData->ovar[_ovari] == 2) {ka = _PP[2];};   if (_solveData->ovar[_ovari] == 3) {mtt = _PP[3];};   if (_solveData->ovar[_ovari] == 4) {f2 = _PP[4];};   if (_solveData->ovar[_ovari] == 5) {n = _PP[5];}; }
#> _getRxSolve_t _getRxSolve_;
#> _simfun simeps;
#> _simfun simeta;
#> _udf_type _evalUdf=NULL;
#> rx_solve *_solveData=NULL;
#> rxode2_assign_ptr _assign_ptr=NULL;
#> _rxRmModelLibType _rxRmModelLib=NULL;
#> _rxGetModelLibType _rxGetModelLib=NULL;
#> rxode2_ode_solver_old_c _old_c=NULL;
#> rxode2_fn0i _ptrid=NULL;
#> _rxIsCurrentC_type _rxIsCurrentC=NULL;
#> _rxSumType _sumPS=NULL;
#> _rxProdType _prodPS=NULL;
#> rxode2_fn0i _prodType=NULL;
#> rxode2_fn0i _sumType=NULL;
#> _update_par_ptr_p _update_par_ptr=NULL;
#> _getParCov_p _getParCov=NULL;
#> linCmtA_p linCmtA;
#> linCmtA_p linCmtC;
#> linCmtB_p linCmtB;
#> _rx_asgn _rxode2_rxAssignPtr=NULL;
#> _rx_asgn _rxQr=NULL;
#> rxode2_fn phi;
#> rxode2_fn ReLU;
#> rxode2_fn dReLU;
#> rxode2_fn GELU;
#> rxode2_fn dGELU;
#> rxode2_fn d2GELU;
#> rxode2_fn d3GELU;
#> rxode2_fn d4GELU;
#> rxode2_fn softplus;
#> rxode2_fn dsoftplus;
#> rxode2_fn d2softplus;
#> rxode2_fn d3softplus;
#> rxode2_fn d4softplus;
#> rxode2_fn SELU;
#> rxode2_fn dSELU;
#> rxode2_fn lReLU;
#> rxode2_fn dlReLU;
#> rxode2_fn Swish;
#> rxode2_fn dSwish;
#> rxode2_fn2 PReLU;
#> rxode2_fn2 dPReLU;
#> rxode2_fn2 dPReLUa;
#> rxode2_fn2 dPReLUa1;
#> rxode2_fn2 ELU;
#> rxode2_fn2 dELU;
#> rxode2_fn2 d2ELU;
#> rxode2_fn2 d2aELU;
#> rxode2_fn2 dELUa;
#> rxode2_fn2 d2ELUa;
#> rxode2_fn3 logit;
#> rxode2_fn3 expit;
#> rxode2_fn2 gammap;
#> rxode2_fn2 gammaq;
#> rxode2_fn2 lowergamma;
#> rxode2_fn2 uppergamma;
#> rxode2_fn2 gammapInv;
#> rxode2_fn2 gammapDer;
#> rxode2_fn2 gammapInva;
#> rxode2_fn2 gammaqInv;
#> rxode2_fn2 gammaqInva;
#> rxode2i_fn2 rxnorm;
#> rxode2i_rxbinom rxbinom;
#> rxode2i_rxbinom rxnbinom;
#> rxode2i_rxbinom rxnbinomMu;
#> rxode2i_fn2 rxcauchy;
#> rxode2i_fn rxchisq;
#> rxode2i_fn rxexp;
#> rxode2i_fn2 rxf;
#> rxode2i_ifn rxgeom;
#> rxode2i_fn2 rxgamma;
#> rxode2i_fn2 rxbeta;
#> rxode2i_ifn rxpois;
#> rxode2i_fn rxt_;
#> rxode2i_fn2 rxunif;
#> rxode2i_fn2 rxweibull;
#> rxode2i2_fn2 rinorm;
#> rxode2i2_ribinom ribinom;
#> rxode2i2_ribinom rinbinom;
#> rxode2i2_ribinom rinbinomMu;
#> rxode2i2_fn2 ricauchy;
#> rxode2i2_fn richisq;
#> rxode2i2_fn riexp;
#> rxode2i2_fn2 rif;
#> rxode2i2_ifn rigeom;
#> rxode2i2_fn2 rigamma;
#> rxode2i2_fn2 ribeta;
#> rxode2i2_ifn ripois;
#> rxode2i2_fn rit_;
#> rxode2i2_fn2 riunif;
#> rxode2i2_fn2 riweibull;
#> rxode2_llikNormFun _llikNorm;
#> rxode2_llikNormFun _llikNormDmean;
#> rxode2_llikNormFun _llikNormDsd;
#> rxode2_llikPoisFun _llikPois;
#> rxode2_llikPoisFun _llikPoisDlambda;
#> rxode2_llikBinomFun _llikBinom;
#> rxode2_llikBinomFun _llikBinomDprob;
#> rxode2_llikBinomFun _llikNbinom;
#> rxode2_llikBinomFun _llikNbinomDprob;
#> rxode2_llikBinomFun _llikNbinomMu;
#> rxode2_llikBinomFun _llikNbinomMuDmu;
#> rxode2_llikBetaFun _llikBeta;
#> rxode2_llikBetaFun _llikBetaDshape1;
#> rxode2_llikBetaFun _llikBetaDshape2;
#> rxode2_llikTFun _llikT;
#> rxode2_llikTFun _llikTDdf;
#> rxode2_llikTFun _llikTDmean;
#> rxode2_llikTFun _llikTDsd;
#> rxode2_llikChisqFun _llikChisq;
#> rxode2_llikChisqFun _llikChisqDdf;
#> rxode2_llikExpFun _llikExp;
#> rxode2_llikExpFun _llikExpDrate;
#> rxode2_llikFFun _llikF;
#> rxode2_llikFFun _llikFDdf1;
#> rxode2_llikFFun _llikFDdf2;
#> rxode2_llikGeomFun _llikGeom;
#> rxode2_llikGeomFun _llikGeomDp;
#> rxode2_llikUnifFun _llikUnif;
#> rxode2_llikUnifFun _llikUnifDalpha;
#> rxode2_llikUnifFun _llikUnifDbeta;
#> rxode2_llikWeibullFun _llikWeibull;
#> rxode2_llikWeibullFun _llikWeibullDshape;
#> rxode2_llikWeibullFun _llikWeibullDscale;
#> rxode2_llikGammaFun _llikGamma;
#> rxode2_llikGammaFun _llikGammaDshape;
#> rxode2_llikGammaFun _llikGammaDrate;
#> rxode2_llikCauchyFun _llikCauchy;
#> rxode2_llikCauchyFun _llikCauchyDlocation;
#> rxode2_llikCauchyFun _llikCauchyDscale;
#> rxode2_compareFactorVal_fn _compareFactorVal;
#> #include "extraC.h"
#> double _prod(double *input, double *p, int type, int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   return _prodPS(input, p, n, type);
#> }
#> double _udf(const char *funName, double *input, int n, ...) {
#>   if (n == -42) Rf_error("%s", "this has a ui user function that cannot be called directly");
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   return _evalUdf(funName, n, input);
#> }
#> double _sum(double *input, double *pld, int m, int type, int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   double ret = _sumPS(input, n, pld, m, type);
#>   if (type == 2 && m < 0){
#>     for (int i = -m; i--;){
#>       pld[i] = 0.0;
#>     }
#>   }
#>   return ret;
#> }
#> double _sign(unsigned int n, ...) {
#>   va_list valist;
#>   va_start(valist, n);
#>   double s = 1;
#>   for (unsigned int i = 0; i < n; i++) {
#>     s = sign(va_arg(valist, double))*s;
#>     if (s == 0){
#>       break;
#>     }
#>   }
#>   va_end(valist);
#>   return s;
#> }
#> double _rxord(int _cSub, unsigned int n,  ...) {
#>   rx_solving_options_ind* ind = &(_solveData->subjects[_cSub]);
#>   if (!ind->inLhs) {
#>     return 1.0;
#>   }
#>   va_list valist;
#>   va_start(valist, n);
#>   double ret = 1.0;
#>   double p = 0.0;
#>   double u = rxunif(ind, 0.0, 1.0);
#>   int found = 0;
#>   for (unsigned int i = 0; i < n; i++) {
#>     p += va_arg(valist, double);
#>     if (!found) {
#>       if (u < p) {
#>         ret = (double)(i+1);
#>         found = 1;
#>       }
#>     }
#>   }
#>   if (!found) ret =(double)(n+1);
#>   va_end(valist);
#>   return ret;
#> }
#> double _max(unsigned int n, ...) {
#>   va_list valist;
#>   va_start(valist, n);
#>   double mx = NA_REAL;
#>   double tmp = 0;
#>   if (n >= 1){
#>     mx = va_arg(valist, double);
#>     for (unsigned int i = 1; i < n; i++) {
#>       tmp = va_arg(valist, double);
#>       if (tmp>mx) mx=tmp;
#>     }
#>     va_end(valist);
#>   }
#>   return mx;
#> }
#> double _min(unsigned int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   double mn = NA_REAL;
#>   double tmp = 0;
#>   if (n >= 1){
#>     mn = va_arg(valist, double);
#>     for (unsigned int i = 1; i < n; i++){
#>       tmp = va_arg(valist, double);
#>       if (tmp<mn) mn=tmp;
#>     }
#>     va_end(valist);
#>   }
#>   return mn;
#> }
#> double _transit4P(int cmt, double t, unsigned int id, double n, double mtt, double bio){
#>   double nd = (double) n;
#>   double ktr = (nd+1)/mtt;
#>   double lktr = _safe_log(nd+1)-_safe_log(mtt);
#>   double tlast = _solveData->subjects[id].tlastS[cmt];
#>   double dose = _solveData->subjects[id].curDoseS[cmt];
#>   if (ISNA(dose)) dose = 0.0;
#>   if (ISNA(tlast)) tlast = 0.0;
#>   double tad = (t-tlast);
#>   return exp(_safe_log(bio*dose)+lktr+n*(lktr+_safe_log(tad))-ktr*(tad)-lgamma1p(nd));
#> }
#> double _transit3P(int cmt, double t, unsigned int id, double n, double mtt){
#>   double nd = (double) n;
#>   double ktr = (nd+1)/mtt;
#>   double lktr = _safe_log(nd+1)-_safe_log(mtt);
#>   double tlast = _solveData->subjects[id].tlastS[cmt];
#>   if (ISNA(tlast)) tlast = 0.0;
#>   double tad = t-tlast;
#>   double podo = _solveData->subjects[id].curDoseS[cmt];
#>   if (ISNA(podo)) podo = 0.0;
#>   return exp(_safe_log(podo)+lktr+n*(lktr+_safe_log(tad))-ktr*(tad)-lgamma1p(nd));
#> }
#> void _assignFuns0(void) {
#>   _evalUdf = (_udf_type) R_GetCCallable("rxode2", "_rxode2_evalUdf");
#>   _getRxSolve_ = (_getRxSolve_t) R_GetCCallable("rxode2","getRxSolve_");
#>   _assign_ptr=(rxode2_assign_ptr) R_GetCCallable("rxode2","rxode2_assign_fn_pointers");
#>   _rxRmModelLib=(_rxRmModelLibType) R_GetCCallable("rxode2","rxRmModelLib");
#>   _rxGetModelLib=(_rxGetModelLibType) R_GetCCallable("rxode2","rxGetModelLib");
#>   _rxode2_rxAssignPtr=(_rx_asgn)R_GetCCallable("rxode2","_rxode2_rxAssignPtr");
#>   _rxQr=(_rx_asgn)R_GetCCallable("rxode2","_rxode2_rxQr");
#>   _rxIsCurrentC = (_rxIsCurrentC_type)R_GetCCallable("rxode2","rxIsCurrentC");
#>   _sumPS  = (_rxSumType) R_GetCCallable("PreciseSums","PreciseSums_sum_r");
#>   _prodPS = (_rxProdType) R_GetCCallable("PreciseSums","PreciseSums_prod_r");
#>   _prodType=(rxode2_fn0i)R_GetCCallable("PreciseSums", "PreciseSums_prod_get");
#>   _sumType=(rxode2_fn0i)R_GetCCallable("PreciseSums", "PreciseSums_sum_get");
#>   _ptrid=(rxode2_fn0i)R_GetCCallable("rxode2", "rxode2_current_fn_pointer_id");
#>   _compareFactorVal=(rxode2_compareFactorVal_fn) R_GetCCallable("rxode2", "compareFactorVal");
#>   _update_par_ptr = (_update_par_ptr_p) R_GetCCallable("rxode2","_update_par_ptr");
#>   _getParCov = (_getParCov_p) R_GetCCallable("rxode2","_getParCov");
#>   // dynamic start
#>   _llikCauchyDscale = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchyDscale");
#>   _llikCauchyDlocation = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchyDlocation");
#>   _llikCauchy = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchy");
#>   _llikGammaDrate = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGammaDrate");
#>   _llikGammaDshape = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGammaDshape");
#>   _llikGamma = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGamma");
#>   _llikWeibullDscale = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibullDscale");
#>   _llikWeibullDshape = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibullDshape");
#>   _llikWeibull = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibull");
#>   _llikUnifDbeta = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnifDbeta");
#>   _llikUnifDalpha = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnifDalpha");
#>   _llikUnif = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnif");
#>   _llikGeomDp = (rxode2_llikGeomFun) R_GetCCallable("rxode2ll", "rxLlikGeomDp");
#>   _llikGeom = (rxode2_llikGeomFun) R_GetCCallable("rxode2ll", "rxLlikGeom");
#>   _llikFDdf2 = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikFDdf2");
#>   _llikFDdf1 = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikFDdf1");
#>   _llikF = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikF");
#>   _llikExpDrate = (rxode2_llikExpFun) R_GetCCallable("rxode2ll", "rxLlikExpDrate");
#>   _llikExp = (rxode2_llikExpFun) R_GetCCallable("rxode2ll", "rxLlikExp");
#>   _llikChisqDdf = (rxode2_llikChisqFun) R_GetCCallable("rxode2ll", "rxLlikChisqDdf");
#>   _llikChisq = (rxode2_llikChisqFun) R_GetCCallable("rxode2ll", "rxLlikChisq");
#>   _llikTDsd = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDsd");
#>   _llikTDmean = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDmean");
#>   _llikTDdf = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDdf");
#>   _llikT = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikT");
#>   _llikBetaDshape2 = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBetaDshape2");
#>   _llikBetaDshape1 = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBetaDshape1");
#>   _llikBeta = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBeta");
#>   _llikNbinomMuDmu = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomMuDmu");
#>   _llikNbinomMu = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomMu");
#>   _llikNbinomDprob = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomDprob");
#>   _llikNbinom = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinom");
#>   _llikBinomDprob = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikBinomDprob");
#>   _llikBinom = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikBinom");
#>   _llikPoisDlambda = (rxode2_llikPoisFun) R_GetCCallable("rxode2ll", "rxLlikPoisDlambda");
#>   _llikPois = (rxode2_llikPoisFun) R_GetCCallable("rxode2ll", "rxLlikPois");
#>   _llikNormDsd = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNormDsd");
#>   _llikNormDmean = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNormDmean");
#>   _llikNorm = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNorm");
#>   simeps = (_simfun) R_GetCCallable("rxode2", "simeps");
#>   simeta = (_simfun) R_GetCCallable("rxode2", "simeta");
#>   expit = (rxode2_fn3) R_GetCCallable("rxode2", "expit");
#>   logit = (rxode2_fn3) R_GetCCallable("rxode2", "logit");
#>   gammapDer = (rxode2_fn2) R_GetCCallable("rxode2", "gammapDer");
#>   lowergamma = (rxode2_fn2) R_GetCCallable("rxode2", "lowergamma");
#>   uppergamma = (rxode2_fn2) R_GetCCallable("rxode2", "uppergamma");
#>   gammaqInva = (rxode2_fn2) R_GetCCallable("rxode2", "gammaqInva");
#>   gammaqInv = (rxode2_fn2) R_GetCCallable("rxode2", "gammaqInv");
#>   gammapInva = (rxode2_fn2) R_GetCCallable("rxode2", "gammapInva");
#>   gammapInv = (rxode2_fn2) R_GetCCallable("rxode2", "gammapInv");
#>   gammaq = (rxode2_fn2) R_GetCCallable("rxode2", "gammaq");
#>   gammap = (rxode2_fn2) R_GetCCallable("rxode2", "gammap");
#>   phi = (rxode2_fn) R_GetCCallable("rxode2", "phi");
#>   d2ELUa = (rxode2_fn2) R_GetCCallable("rxode2", "d2ELUa");
#>   dELUa = (rxode2_fn2) R_GetCCallable("rxode2", "dELUa");
#>   d2aELU = (rxode2_fn2) R_GetCCallable("rxode2", "d2aELU");
#>   d2ELU = (rxode2_fn2) R_GetCCallable("rxode2", "d2ELU");
#>   dELU = (rxode2_fn2) R_GetCCallable("rxode2", "dELU");
#>   ELU = (rxode2_fn2) R_GetCCallable("rxode2", "ELU");
#>   dPReLUa1 = (rxode2_fn2) R_GetCCallable("rxode2", "dPReLUa1");
#>   dPReLUa = (rxode2_fn2) R_GetCCallable("rxode2", "dPReLUa");
#>   dPReLU = (rxode2_fn2) R_GetCCallable("rxode2", "dPReLU");
#>   PReLU = (rxode2_fn2) R_GetCCallable("rxode2", "PReLU");
#>   dSwish = (rxode2_fn) R_GetCCallable("rxode2", "dSwish");
#>   Swish = (rxode2_fn) R_GetCCallable("rxode2", "Swish");
#>   dlReLU = (rxode2_fn) R_GetCCallable("rxode2", "dlReLU");
#>   lReLU = (rxode2_fn) R_GetCCallable("rxode2", "lReLU");
#>   dSELU = (rxode2_fn) R_GetCCallable("rxode2", "dSELU");
#>   SELU = (rxode2_fn) R_GetCCallable("rxode2", "SELU");
#>   d4softplus = (rxode2_fn) R_GetCCallable("rxode2", "d4softplus");
#>   d3softplus = (rxode2_fn) R_GetCCallable("rxode2", "d3softplus");
#>   d2softplus = (rxode2_fn) R_GetCCallable("rxode2", "d2softplus");
#>   dsoftplus = (rxode2_fn) R_GetCCallable("rxode2", "dsoftplus");
#>   softplus = (rxode2_fn) R_GetCCallable("rxode2", "softplus");
#>   d4GELU = (rxode2_fn) R_GetCCallable("rxode2", "d4GELU");
#>   d3GELU = (rxode2_fn) R_GetCCallable("rxode2", "d3GELU");
#>   d2GELU = (rxode2_fn) R_GetCCallable("rxode2", "d2GELU");
#>   dGELU = (rxode2_fn) R_GetCCallable("rxode2", "dGELU");
#>   GELU = (rxode2_fn) R_GetCCallable("rxode2", "GELU");
#>   dReLU = (rxode2_fn) R_GetCCallable("rxode2", "dReLU");
#>   ReLU = (rxode2_fn) R_GetCCallable("rxode2", "ReLU");
#>   riweibull = (rxode2i2_fn2) R_GetCCallable("rxode2", "riweibull");
#>   riunif = (rxode2i2_fn2) R_GetCCallable("rxode2", "riunif");
#>   rit_ = (rxode2i2_fn) R_GetCCallable("rxode2", "rit_");
#>   ripois = (rxode2i2_ifn) R_GetCCallable("rxode2", "ripois");
#>   ribeta = (rxode2i2_fn2) R_GetCCallable("rxode2", "ribeta");
#>   rigamma = (rxode2i2_fn2) R_GetCCallable("rxode2", "rigamma");
#>   rigeom = (rxode2i2_ifn) R_GetCCallable("rxode2", "rigeom");
#>   rif = (rxode2i2_fn2) R_GetCCallable("rxode2", "rif");
#>   riexp = (rxode2i2_fn) R_GetCCallable("rxode2", "riexp");
#>   richisq = (rxode2i2_fn) R_GetCCallable("rxode2", "richisq");
#>   ricauchy = (rxode2i2_fn2) R_GetCCallable("rxode2", "ricauchy");
#>   rinbinomMu = (rxode2i2_ribinom) R_GetCCallable("rxode2", "rinbinomMu");
#>   rinbinom = (rxode2i2_ribinom) R_GetCCallable("rxode2", "rinbinom");
#>   ribinom = (rxode2i2_ribinom) R_GetCCallable("rxode2", "ribinom");
#>   rinorm = (rxode2i2_fn2) R_GetCCallable("rxode2", "rinorm");
#>   rxweibull = (rxode2i_fn2) R_GetCCallable("rxode2", "rxweibull");
#>   rxunif = (rxode2i_fn2) R_GetCCallable("rxode2", "rxunif");
#>   rxt_ = (rxode2i_fn) R_GetCCallable("rxode2", "rxt_");
#>   rxpois = (rxode2i_ifn) R_GetCCallable("rxode2", "rxpois");
#>   rxbeta = (rxode2i_fn2) R_GetCCallable("rxode2", "rxbeta");
#>   rxgamma = (rxode2i_fn2) R_GetCCallable("rxode2", "rxgamma");
#>   rxgeom = (rxode2i_ifn) R_GetCCallable("rxode2", "rxgeom");
#>   rxf = (rxode2i_fn2) R_GetCCallable("rxode2", "rxf");
#>   rxexp = (rxode2i_fn) R_GetCCallable("rxode2", "rxexp");
#>   rxchisq = (rxode2i_fn) R_GetCCallable("rxode2", "rxchisq");
#>   rxcauchy = (rxode2i_fn2) R_GetCCallable("rxode2", "rxcauchy");
#>   rxnbinomMu = (rxode2i_rxbinom) R_GetCCallable("rxode2", "rxnbinomMu");
#>   rxnbinom = (rxode2i_rxbinom) R_GetCCallable("rxode2", "rxnbinom");
#>   rxbinom = (rxode2i_rxbinom) R_GetCCallable("rxode2", "rxbinom");
#>   rxnorm = (rxode2i_fn2) R_GetCCallable("rxode2", "rxnorm");
#>   linCmtC = (linCmtA_p) R_GetCCallable("rxode2", "linCmtC");
#>   linCmtB = (linCmtB_p) R_GetCCallable("rxode2", "linCmtB");
#>   linCmtA = (linCmtA_p) R_GetCCallable("rxode2", "linCmtA");
#>   // dynamic stop
#>   _solveData = _getRxSolve_();
#> }
#> void _assignFuns(void) {
#>   if (_assign_ptr == NULL){
#>     _assignFuns0();
#>   }
#> }
#> void __assignFuns2(rx_solve rx,
#>                    rx_solving_options op,
#>                    t_F f,
#>                    t_LAG lag,
#>                    t_RATE rate,
#>                    t_DUR dur,
#>                    t_calc_mtime mtime,
#>                    t_ME me,
#>                    t_IndF indf,
#>                    t_getTime gettime,
#>                    t_locateTimeIndex timeindex,
#>                    t_handle_evidL handleEvid,
#>                    t_getDur getdur) {
#>   // assign start
#>     static rxode2_assignFuns2_t rxode2_assignFuns2 = NULL;
#>     if (rxode2_assignFuns2 == NULL) rxode2_assignFuns2 = (rxode2_assignFuns2_t)(R_GetCCallable("rxode2", "_rxode2_assignFuns2"));
#>     rxode2_assignFuns2(rx, op, f, lag, rate, dur, mtime, me, indf, gettime, timeindex, handleEvid, getdur);
#>   // assign stop
#> }
#> extern void  rx_c1b42db8d18f7ac7822d3b8d3093b99f__ode_solver_solvedata (rx_solve *solve){
#>   _solveData = solve;
#> }
#> extern rx_solve *rx_c1b42db8d18f7ac7822d3b8d3093b99f__ode_solver_get_solvedata(void){
#>   return _solveData;
#> }
#> SEXP rx_c1b42db8d18f7ac7822d3b8d3093b99f__model_vars(void);
#> 
#> 
#> // prj-specific differential eqns
#> void rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt(int *_neq, double __t, double *__zzStateVar__, double *__DDtStateVar__)
#> {
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   int _cSub = _neq[1];
#>   double t = __t + _solveData->subjects[_neq[1]].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=1;
#>     double cl = NA_REAL;
#>   double vc = NA_REAL;
#>   double ka = NA_REAL;
#>   double mtt = NA_REAL;
#>   double f2 = NA_REAL;
#>   double n = NA_REAL;
#>   double k = NA_REAL;
#>   double bio = NA_REAL;
#>   double ktr = NA_REAL;
#>   double depot1 = NA_REAL;
#>   double depot2 = NA_REAL;
#>   double cen = NA_REAL;
#> 
#>   (void)t;
#>   (void)cl;
#>   (void)vc;
#>   (void)ka;
#>   (void)mtt;
#>   (void)f2;
#>   (void)n;
#>   (void)k;
#>   (void)bio;
#>   (void)ktr;
#>   (void)depot1;
#>   (void)depot2;
#>   (void)cen;
#> 
#>   k = _PL[0];
#>   bio = _PL[1];
#>   ktr = _PL[2];
#> 
#>   _update_par_ptr(__t, _cSub, _solveData, _idx);
#>   cl = _PP[0];
#>   vc = _PP[1];
#>   ka = _PP[2];
#>   mtt = _PP[3];
#>   f2 = _PP[4];
#>   n = _PP[5];
#> 
#>   depot1 = __zzStateVar__[__DDT0__]*((double)(_ON[__DDT0__]));
#>   depot2 = __zzStateVar__[__DDT1__]*((double)(_ON[__DDT1__]));
#>   cen = __zzStateVar__[__DDT2__]*((double)(_ON[__DDT2__]));
#> 
#>   k =cl/_div0(vc);
#>   bio =1-f2;
#>   ktr=(n+1)/_div0(mtt);
#>   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>   __DDtStateVar__[__DDT1__] = ((double)(_ON[__DDT1__]))*(_IR[__DDT1__] -ka*depot2);
#>   __DDtStateVar__[__DDT2__] = ((double)(_ON[__DDT2__]))*(_IR[__DDT2__] + ka*depot1+ka*depot2-k*cen);
#>   (&_solveData->subjects[_cSub])->dadt_counter[0]++;
#> }
#> 
#> // Jacobian derived vars
#> void rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_jac(int *_neq, double __t, double *__zzStateVar__, double *__PDStateVar__, unsigned int __NROWPD__) {
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>     int _cSub=_neq[1];
#>   double t = __t + _solveData->subjects[_neq[1]].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=2;
#>     (&_solveData->subjects[_cSub])->jac_counter[0]++;
#> }
#> // Functional based initial conditions.
#> void rx_c1b42db8d18f7ac7822d3b8d3093b99f__inis(int _cSub, double *__zzStateVar__){
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   
#>   (&_solveData->subjects[_cSub])->_rxFlag=3;
#>   }
#> // prj-specific derived vars
#> void rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_lhs(int _cSub, double __t, double *__zzStateVar__, double *_lhs) {
#>     int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=11;
#>     double  __DDtStateVar_0__;
#>   double  __DDtStateVar_1__;
#>   double  __DDtStateVar_2__;
#>   double cl = NA_REAL;
#>   double vc = NA_REAL;
#>   double ka = NA_REAL;
#>   double mtt = NA_REAL;
#>   double f2 = NA_REAL;
#>   double n = NA_REAL;
#>   double k = NA_REAL;
#>   double bio = NA_REAL;
#>   double ktr = NA_REAL;
#>   double depot1 = NA_REAL;
#>   double depot2 = NA_REAL;
#>   double cen = NA_REAL;
#> 
#>   (void)t;
#>   (void)__DDtStateVar_0__;
#>   (void)__DDtStateVar_1__;
#>   (void)__DDtStateVar_2__;
#>   (void)cl;
#>   (void)vc;
#>   (void)ka;
#>   (void)mtt;
#>   (void)f2;
#>   (void)n;
#>   (void)k;
#>   (void)bio;
#>   (void)ktr;
#>   (void)depot1;
#>   (void)depot2;
#>   (void)cen;
#> 
#>   k = _PL[0];
#>   bio = _PL[1];
#>   ktr = _PL[2];
#> 
#>   _update_par_ptr(__t, _cSub, _solveData, _idx);
#>   cl = _PP[0];
#>   vc = _PP[1];
#>   ka = _PP[2];
#>   mtt = _PP[3];
#>   f2 = _PP[4];
#>   n = _PP[5];
#> 
#>   depot1 = __zzStateVar__[__DDT0__]*((double)(_ON[__DDT0__]));
#>   depot2 = __zzStateVar__[__DDT1__]*((double)(_ON[__DDT1__]));
#>   cen = __zzStateVar__[__DDT2__]*((double)(_ON[__DDT2__]));
#> 
#>   k =cl/_div0(vc);
#>   bio =1-f2;
#>   ktr=(n+1)/_div0(mtt);
#>   __DDtStateVar_0__ = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] + exp(_safe_log(bio*_podo1(_DEPOT_))+_safe_log(ktr)+n*_safe_log(ktr*_tad1(_DEPOT_))-ktr*_tad1(_DEPOT_)-lgammafn(n+1))-ka*depot1);
#>   __DDtStateVar_1__ = ((double)(_ON[__DDT1__]))*(_IR[__DDT1__] -ka*depot2);
#>   __DDtStateVar_2__ = ((double)(_ON[__DDT2__]))*(_IR[__DDT2__] + ka*depot1+ka*depot2-k*cen);
#> 
#>   _lhs[0]=k;
#>   _lhs[1]=bio;
#>   _lhs[2]=ktr;
#> }
#> // Functional based bioavailability (returns amount)
#> double rx_c1b42db8d18f7ac7822d3b8d3093b99f__F(int _cSub,  int _cmt, double _amt, double __t, double *__zzStateVar__){
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   double *_f=_solveData->subjects[_cSub].cF;
#>   (void)_f;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=4;
#>     _f[2]=1.0;
#>   _f[1]=1.0;
#>   _f[0]=1.0;
#>   double cl = NA_REAL;
#>   double vc = NA_REAL;
#>   double ka = NA_REAL;
#>   double mtt = NA_REAL;
#>   double f2 = NA_REAL;
#>   double n = NA_REAL;
#>   double k = NA_REAL;
#>   double bio = NA_REAL;
#>   double ktr = NA_REAL;
#>   double depot1 = NA_REAL;
#>   double depot2 = NA_REAL;
#>   double cen = NA_REAL;
#> 
#>   (void)t;
#>   (void)cl;
#>   (void)vc;
#>   (void)ka;
#>   (void)mtt;
#>   (void)f2;
#>   (void)n;
#>   (void)k;
#>   (void)bio;
#>   (void)ktr;
#>   (void)depot1;
#>   (void)depot2;
#>   (void)cen;
#> 
#>   k = _PL[0];
#>   bio = _PL[1];
#>   ktr = _PL[2];
#> 
#>   _update_par_ptr(__t, _cSub, _solveData, _idx);
#>   cl = _PP[0];
#>   vc = _PP[1];
#>   ka = _PP[2];
#>   mtt = _PP[3];
#>   f2 = _PP[4];
#>   n = _PP[5];
#> 
#>   depot1 = __zzStateVar__[__DDT0__]*((double)(_ON[__DDT0__]));
#>   depot2 = __zzStateVar__[__DDT1__]*((double)(_ON[__DDT1__]));
#>   cen = __zzStateVar__[__DDT2__]*((double)(_ON[__DDT2__]));
#> 
#>   k =cl/_div0(vc);
#>   bio =1-f2;
#>   ktr=(n+1)/_div0(mtt);
#>   _f[__DDT1__] = f2;
#> 
#>   return _f[_cmt]*_amt;
#> }
#> // Functional based absorption lag
#> double rx_c1b42db8d18f7ac7822d3b8d3093b99f__Lag(int _cSub,  int _cmt, double __t, double *__zzStateVar__){
#>  return __t;
#> }
#> // Modeled zero-order rate
#> double rx_c1b42db8d18f7ac7822d3b8d3093b99f__Rate(int _cSub,  int _cmt, double _amt, double __t, double *__zzStateVar__){
#>  return 0.0;
#> }
#> // Modeled zero-order duration
#> double rx_c1b42db8d18f7ac7822d3b8d3093b99f__Dur(int _cSub,  int _cmt, double _amt, double __t){
#>  return 0.0;
#> }
#> // Model Times
#> void rx_c1b42db8d18f7ac7822d3b8d3093b99f__mtime(int _cSub, double *_mtime){
#> }
#> // Matrix Exponential (0)
#> void rx_c1b42db8d18f7ac7822d3b8d3093b99f__ME(int _cSub, double _t, double __t, double *_mat, const double *__zzStateVar__){
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=9;
#>   }
#> // Inductive linearization Matf
#> void rx_c1b42db8d18f7ac7822d3b8d3093b99f__IndF(int _cSub, double _t, double __t, double *_matf){
#>  int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=10;
#>   }
#> extern SEXP rx_c1b42db8d18f7ac7822d3b8d3093b99f__model_vars(void){
#>   int pro=0;
#>   SEXP _mv = PROTECT(_rxGetModelLib("rx_c1b42db8d18f7ac7822d3b8d3093b99f__model_vars"));pro++;
#>   if (!_rxIsCurrentC(_mv)){
#>     SEXP hash    = PROTECT(Rf_allocVector(STRSXP, 1));pro++;
#> #define __doBuf__  snprintf(buf, __doBufN__, "un]\"BAAA@QRtHACAAAAAAAN/!AAAv7#aT),vKyPAgkP<gtjmYB0E1U6S0IpYU!J}pwKi;*<!dwWmvO7VS$H6+mb~q5V{r{BK7*6^4Q!_rab$h\?`Hy#.U@Q`hsu;b6y.>TA,GG;@!5VhK4%%J*O!aCB5X&DC_e3$V>p(.2s10h}.diq%%4@^m]|#G#>T{C+K_u6M$Fx[RQ=]Oc{[:y!IgfUV<]11OR$FT,J&G2VoGf6h+3c3dS+s;%%XAz8#(\?M#\?\?.UC3c$y5@oX/c2h[,m2_t*wUrd^DK\?fOf:iUg05ULLY00V\?>nUHL&cl~QN[CZ@qi6LJRoI!}M[tc%%P&jwmQm{EJ+TcS+cF4r/l&M\"8|nPSfN`EvgkkOCoG),OV}[M1214JwCK%%37khzkEzA}3)sMI]vI;]>0@L2g4f`!4HQSW%%2&cYGo{~1@`H\"le5K\?M1!d2Ezg+VewocoM<>M1:T,xP`<JXuJYKM(/npmF<L4uwN#k%%tD:NKqS+`pZpq=5M*A<zDy;^f3UCp1vS5KXz/WmAl6#\"Th051umLUT$6._[KeD^t$UHEwLB+T!\"UTs1A{`8\?e:PO9S;>=J*,(j|8QL~]B/#y~$i`S5U|l`sG@70,>>UMXl~x\?sHemlB&t>WlB1s;(by)tYYtDkPwvKrq.VgXot/TajGlVQpD*<G+POk\",MyU:kb]0mZOaIFR$]1./Smt3^l3ObU5\?q5@;SHvBK+(p7[FZyizzpD]u_fRQ(>v;ued43}R7~Kteco}[Tg6#r!t:z[a0!86+<H1~q)k(MlOBOrn&FUd!lwSZTpj~;NPmh%%LU=5jdW9Z7M^Gc|&!TG910i8*yTg>X:GxTMly7wv[1#.~+TVv@MU$RNg=IK[UHHPg2>.&w\"Lc|\?XkEv_AALH5AGO,`]K}Vvg+XQr#q~I2Tfv&j*1B22Z=Z`H%%ZO)2W/FJGTyyW~:N*[F5LZX)tlZUwlGXjAjtS9b3fF^ks/9U@noDR^v^n0v{F.RILnT:l)){}1E$OrFQo:K+X5Q5PK,XBx]5)KUb`bw)PId[;_N:DKXFEWh7yI_8O/7#SRC|4<`@nZwE.}O`&vsHVEfob3(OOo@y+U7>SCPm$y=Q3})TKmC^EGs5icwp*m/>!j]BepTxj7]_6~!6P>P>.ca2_R7+Uy^lR@|f4F!PP^9&E=F*sNe|m,(/V]D\"xV2bP/bU`aIiN,3%%7{eN~TOCzM;^QHe&1nC)D.]`6+cYv\?aWw(1NMus5Yim,Wi,t/i@;$8znRkT<eu/o~UECa\?");
#>     char buf[1280];
#> #define __doBufN__ 1280
#>     __doBuf__
#> #undef __doBuf__
#> #undef __doBufN__
#>     SET_STRING_ELT(hash, 0, mkChar(buf));
#>     SEXP lst      = PROTECT(_rxQr(hash));pro++;
#>     _assign_ptr(lst);
#>     UNPROTECT(pro);
#>     return lst;
#>   } else {
#>     UNPROTECT(pro);
#>     return _mv;
#>   }
#> }
#> extern void rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt_lsoda(int *neq, double *t, double *A, double *DADT)
#> {
#>   rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt(neq, *t, A, DADT);
#> }
#> extern int rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt_liblsoda(double __t, double *y, double *ydot, void *data)
#> {
#>   int *neq = (int*)(data);
#>   rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt(neq, __t, y, ydot);
#>   return(0);
#> }
#> extern void rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_jac_lsoda(int *neq, double *t, double *A,int *ml, int *mu, double *JAC, int *nrowpd){
#>   // Update all covariate parameters
#>   rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_jac(neq, *t, A, JAC, *nrowpd);
#> }
#> 
#> //Create function to call from R's main thread that assigns the required functions. Sometimes they don't get assigned.
#> extern void rx_c1b42db8d18f7ac7822d3b8d3093b99f__assignFuns(void){
#>   _assignFuns();
#> }
#> 
#> //Initialize the dll to match rxode2's calls
#> void R_init0_rx_c1b42db8d18f7ac7822d3b8d3093b99f_(void){
#>   // Get C callables on load; Otherwise it isn't thread safe
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__assignFuns2", (DL_FUNC) __assignFuns2);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__assignFuns", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__assignFuns);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__inis",(DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__inis);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt",(DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_lhs",(DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_lhs);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_jac",(DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_jac);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt_lsoda", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt_lsoda);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_jac_lsoda", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__calc_jac_lsoda);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__ode_solver_solvedata", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__ode_solver_solvedata);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__ode_solver_get_solvedata", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__ode_solver_get_solvedata);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__F", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__F);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__Lag", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__Lag);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__Rate", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__Rate);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__Dur", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__Dur);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__mtime", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__mtime);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__ME", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__ME);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__IndF", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__IndF);
#>   R_RegisterCCallable("rx_c1b42db8d18f7ac7822d3b8d3093b99f_","rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt_liblsoda", (DL_FUNC) rx_c1b42db8d18f7ac7822d3b8d3093b99f__dydt_liblsoda);
#> }
#> //Initialize the dll to match rxode2's calls
#> void R_init_rx_c1b42db8d18f7ac7822d3b8d3093b99f_(DllInfo *info){
#>   // Get C callables on load; Otherwise it isn't thread safe
#>   R_init0_rx_c1b42db8d18f7ac7822d3b8d3093b99f_();
#>   static const R_CallMethodDef callMethods[]  = {
#>     {"rx_c1b42db8d18f7ac7822d3b8d3093b99f__model_vars", (DL_FUNC) &rx_c1b42db8d18f7ac7822d3b8d3093b99f__model_vars, 0},
#>     {NULL, NULL, 0}
#>   };
#> 
#>   R_registerRoutines(info, NULL, callMethods, NULL, NULL);
#>   R_useDynamicSymbols(info,FALSE);
#>   _assignFuns0();
#> 
#> }
#> 
#> void R_unload_rx_c1b42db8d18f7ac7822d3b8d3093b99f_ (DllInfo *info){
#>   // Free resources required for single subject solve.
#>   SEXP _mv = PROTECT(_rxGetModelLib("rx_c1b42db8d18f7ac7822d3b8d3093b99f__model_vars"));
#>   if (!isNull(_mv)){
#>     _rxRmModelLib("rx_c1b42db8d18f7ac7822d3b8d3093b99f__model_vars");
#>   }
#>   UNPROTECT(1);
#> }
#> Error: error building model
```

<sup>Created on 2024-11-29 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Related to #804

Also this should work:

``` r
library(rxode2)
#> rxode2 3.0.2.9000 using 8 threads (see ?getRxThreads)
#>   no cache: create with `rxCreateCache()`
mod = function ()
{
  description <- "One compartment PK model with linear clearance using differential equations"
  dosing <- c("central", "depot")
  ini({
    lka <- 0.45
    label("Absorption rate (Ka)")
    lcl <- 1
    label("Clearance (CL)")
    lvc <- 3.45
    label("Central volume of distribution (V)")
    propSd <- c(0, 0.5)
    label("Proportional residual error (fraction)")
  })
  model({
    ka <- exp(lka)
    cl <- exp(lcl)
    vc <- exp(lvc)
    kel <- cl/vc
    # you need to put your time-dependent logic here or use a covariate in
    # your dataset
    ka_t <-ka*(1+exp(-.1*tad0(depot)))
    d/dt(depot) <- -ka_t * depot
    d/dt(central) <- ka_t * depot - kel * central
    Cc <- central/vc
    Cc ~ prop(propSd)
  })
}

mod <- mod()

mod$simulationModel
#> using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’
#> In file included from /usr/share/R/include/R.h:69,
#>                  from /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2parse.h:40,
#>                  from /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2.h:10,
#>                  from /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:3,
#>                  from rx_3ad488a46759958958c15fcf7a453770_.c:149:
#> rx_3ad488a46759958958c15fcf7a453770_.c: In function ‘rx_3ad488a46759958958c15fcf7a453770__dydt’:
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:31: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                               ^~~~~~~
#> /usr/share/R/include/R_ext/Arith.h:64:31: note: in definition of macro ‘ISNA’
#>    64 | #define ISNA(x)        R_IsNA(x)
#>       |                               ^
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:24: note: in expansion of macro ‘_tad01’
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                        ^~~~~~
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:31: note: each undeclared identifier is reported only once for each function it appears in
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                               ^~~~~~~
#> /usr/share/R/include/R_ext/Arith.h:64:31: note: in definition of macro ‘ISNA’
#>    64 | #define ISNA(x)        R_IsNA(x)
#>       |                               ^
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:24: note: in expansion of macro ‘_tad01’
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                        ^~~~~~
#> rx_3ad488a46759958958c15fcf7a453770_.c: In function ‘rx_3ad488a46759958958c15fcf7a453770__calc_lhs’:
#> rx_3ad488a46759958958c15fcf7a453770_.c:773:31: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   773 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                               ^~~~~~~
#> /usr/share/R/include/R_ext/Arith.h:64:31: note: in definition of macro ‘ISNA’
#>    64 | #define ISNA(x)        R_IsNA(x)
#>       |                               ^
#> rx_3ad488a46759958958c15fcf7a453770_.c:773:24: note: in expansion of macro ‘_tad01’
#>   773 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                        ^~~~~~
#> make: *** [/usr/lib/R/etc/Makeconf:195: rx_3ad488a46759958958c15fcf7a453770_.o] Error 1
#> error building model
#> ccache gcc -I"/usr/share/R/include" -DNDEBUG      -O3  -I"/home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include" -I"/home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include" -fpic  -O0 -Wall -c rx_3ad488a46759958958c15fcf7a453770_.c -o rx_3ad488a46759958958c15fcf7a453770_.o
#> using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’
#> In file included from /usr/share/R/include/R.h:69,
#>                  from /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2parse.h:40,
#>                  from /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2.h:10,
#>                  from /home/matt/R/x86_64-pc-linux-gnu-library/4.4/rxode2/include/rxode2_model_shared.h:3,
#>                  from rx_3ad488a46759958958c15fcf7a453770_.c:149:
#> rx_3ad488a46759958958c15fcf7a453770_.c: In function ‘rx_3ad488a46759958958c15fcf7a453770__dydt’:
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:31: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                               ^~~~~~~
#> /usr/share/R/include/R_ext/Arith.h:64:31: note: in definition of macro ‘ISNA’
#>    64 | #define ISNA(x)        R_IsNA(x)
#>       |                               ^
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:24: note: in expansion of macro ‘_tad01’
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                        ^~~~~~
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:31: note: each undeclared identifier is reported only once for each function it appears in
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                               ^~~~~~~
#> /usr/share/R/include/R_ext/Arith.h:64:31: note: in definition of macro ‘ISNA’
#>    64 | #define ISNA(x)        R_IsNA(x)
#>       |                               ^
#> rx_3ad488a46759958958c15fcf7a453770_.c:657:24: note: in expansion of macro ‘_tad01’
#>   657 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                        ^~~~~~
#> rx_3ad488a46759958958c15fcf7a453770_.c: In function ‘rx_3ad488a46759958958c15fcf7a453770__calc_lhs’:
#> rx_3ad488a46759958958c15fcf7a453770_.c:773:31: error: ‘_DEPOT_’ undeclared (first use in this function)
#>   773 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                               ^~~~~~~
#> /usr/share/R/include/R_ext/Arith.h:64:31: note: in definition of macro ‘ISNA’
#>    64 | #define ISNA(x)        R_IsNA(x)
#>       |                               ^
#> rx_3ad488a46759958958c15fcf7a453770_.c:773:24: note: in expansion of macro ‘_tad01’
#>   773 |   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>       |                        ^~~~~~
#> make: *** [/usr/lib/R/etc/Makeconf:195: rx_3ad488a46759958958c15fcf7a453770_.o] Error 1
#> // Define translation state order for 3 states
#> #define __DDT0__ 0 // depot
#> #define __DDT1__ 1 // central
#> #define __DDT2__ 2 // NA
#> #define _getRxSolve_ _rxa5ea762553835a063db838c7489b66b80
#> #define _evalUdf _rxa5ea762553835a063db838c7489b66b81
#> #define _solveData _rxa5ea762553835a063db838c7489b66b82
#> #define _assign_ptr _rxa5ea762553835a063db838c7489b66b83
#> #define _rxRmModelLib _rxa5ea762553835a063db838c7489b66b84
#> #define _rxGetModelLib _rxa5ea762553835a063db838c7489b66b85
#> #define _old_c _rxa5ea762553835a063db838c7489b66b86
#> #define _ptrid _rxa5ea762553835a063db838c7489b66b87
#> #define _rxIsCurrentC _rxa5ea762553835a063db838c7489b66b88
#> #define _sumPS _rxa5ea762553835a063db838c7489b66b89
#> #define _prodPS _rxa5ea762553835a063db838c7489b66b810
#> #define _prodType _rxa5ea762553835a063db838c7489b66b811
#> #define _sumType _rxa5ea762553835a063db838c7489b66b812
#> #define _update_par_ptr _rxa5ea762553835a063db838c7489b66b813
#> #define _getParCov _rxa5ea762553835a063db838c7489b66b814
#> #define _rxode2_rxAssignPtr _rxa5ea762553835a063db838c7489b66b815
#> #define _rxQr _rxa5ea762553835a063db838c7489b66b816
#> #define _compareFactorVal _rxa5ea762553835a063db838c7489b66b817
#> #define _sum _rxa5ea762553835a063db838c7489b66b818
#> #define _udf _rxa5ea762553835a063db838c7489b66b819
#> #define _sign _rxa5ea762553835a063db838c7489b66b820
#> #define _prod _rxa5ea762553835a063db838c7489b66b821
#> #define _max _rxa5ea762553835a063db838c7489b66b822
#> #define _min _rxa5ea762553835a063db838c7489b66b823
#> #define _transit4P _rxa5ea762553835a063db838c7489b66b824
#> #define _transit3P _rxa5ea762553835a063db838c7489b66b825
#> #define _assignFuns0 _rxa5ea762553835a063db838c7489b66b826
#> #define _assignFuns _rxa5ea762553835a063db838c7489b66b827
#> #define _rxord _rxa5ea762553835a063db838c7489b66b828
#> #define __assignFuns2 _rxa5ea762553835a063db838c7489b66b829
#> #define _llikCauchyDscale _rxa5ea762553835a063db838c7489b66b830
#> #define _llikCauchyDlocation _rxa5ea762553835a063db838c7489b66b831
#> #define _llikCauchy _rxa5ea762553835a063db838c7489b66b832
#> #define _llikGammaDrate _rxa5ea762553835a063db838c7489b66b833
#> #define _llikGammaDshape _rxa5ea762553835a063db838c7489b66b834
#> #define _llikGamma _rxa5ea762553835a063db838c7489b66b835
#> #define _llikWeibullDscale _rxa5ea762553835a063db838c7489b66b836
#> #define _llikWeibullDshape _rxa5ea762553835a063db838c7489b66b837
#> #define _llikWeibull _rxa5ea762553835a063db838c7489b66b838
#> #define _llikUnifDbeta _rxa5ea762553835a063db838c7489b66b839
#> #define _llikUnifDalpha _rxa5ea762553835a063db838c7489b66b840
#> #define _llikUnif _rxa5ea762553835a063db838c7489b66b841
#> #define _llikGeomDp _rxa5ea762553835a063db838c7489b66b842
#> #define _llikGeom _rxa5ea762553835a063db838c7489b66b843
#> #define _llikFDdf2 _rxa5ea762553835a063db838c7489b66b844
#> #define _llikFDdf1 _rxa5ea762553835a063db838c7489b66b845
#> #define _llikF _rxa5ea762553835a063db838c7489b66b846
#> #define _llikExpDrate _rxa5ea762553835a063db838c7489b66b847
#> #define _llikExp _rxa5ea762553835a063db838c7489b66b848
#> #define _llikChisqDdf _rxa5ea762553835a063db838c7489b66b849
#> #define _llikChisq _rxa5ea762553835a063db838c7489b66b850
#> #define _llikTDsd _rxa5ea762553835a063db838c7489b66b851
#> #define _llikTDmean _rxa5ea762553835a063db838c7489b66b852
#> #define _llikTDdf _rxa5ea762553835a063db838c7489b66b853
#> #define _llikT _rxa5ea762553835a063db838c7489b66b854
#> #define _llikBetaDshape2 _rxa5ea762553835a063db838c7489b66b855
#> #define _llikBetaDshape1 _rxa5ea762553835a063db838c7489b66b856
#> #define _llikBeta _rxa5ea762553835a063db838c7489b66b857
#> #define _llikNbinomMuDmu _rxa5ea762553835a063db838c7489b66b858
#> #define _llikNbinomMu _rxa5ea762553835a063db838c7489b66b859
#> #define _llikNbinomDprob _rxa5ea762553835a063db838c7489b66b860
#> #define _llikNbinom _rxa5ea762553835a063db838c7489b66b861
#> #define _llikBinomDprob _rxa5ea762553835a063db838c7489b66b862
#> #define _llikBinom _rxa5ea762553835a063db838c7489b66b863
#> #define _llikPoisDlambda _rxa5ea762553835a063db838c7489b66b864
#> #define _llikPois _rxa5ea762553835a063db838c7489b66b865
#> #define _llikNormDsd _rxa5ea762553835a063db838c7489b66b866
#> #define _llikNormDmean _rxa5ea762553835a063db838c7489b66b867
#> #define _llikNorm _rxa5ea762553835a063db838c7489b66b868
#> #define simeps _rxa5ea762553835a063db838c7489b66b869
#> #define simeta _rxa5ea762553835a063db838c7489b66b870
#> #define expit _rxa5ea762553835a063db838c7489b66b871
#> #define logit _rxa5ea762553835a063db838c7489b66b872
#> #define gammapDer _rxa5ea762553835a063db838c7489b66b873
#> #define lowergamma _rxa5ea762553835a063db838c7489b66b874
#> #define uppergamma _rxa5ea762553835a063db838c7489b66b875
#> #define gammaqInva _rxa5ea762553835a063db838c7489b66b876
#> #define gammaqInv _rxa5ea762553835a063db838c7489b66b877
#> #define gammapInva _rxa5ea762553835a063db838c7489b66b878
#> #define gammapInv _rxa5ea762553835a063db838c7489b66b879
#> #define gammaq _rxa5ea762553835a063db838c7489b66b880
#> #define gammap _rxa5ea762553835a063db838c7489b66b881
#> #define phi _rxa5ea762553835a063db838c7489b66b882
#> #define d2ELUa _rxa5ea762553835a063db838c7489b66b883
#> #define dELUa _rxa5ea762553835a063db838c7489b66b884
#> #define d2aELU _rxa5ea762553835a063db838c7489b66b885
#> #define d2ELU _rxa5ea762553835a063db838c7489b66b886
#> #define dELU _rxa5ea762553835a063db838c7489b66b887
#> #define ELU _rxa5ea762553835a063db838c7489b66b888
#> #define dPReLUa1 _rxa5ea762553835a063db838c7489b66b889
#> #define dPReLUa _rxa5ea762553835a063db838c7489b66b890
#> #define dPReLU _rxa5ea762553835a063db838c7489b66b891
#> #define PReLU _rxa5ea762553835a063db838c7489b66b892
#> #define dSwish _rxa5ea762553835a063db838c7489b66b893
#> #define Swish _rxa5ea762553835a063db838c7489b66b894
#> #define dlReLU _rxa5ea762553835a063db838c7489b66b895
#> #define lReLU _rxa5ea762553835a063db838c7489b66b896
#> #define dSELU _rxa5ea762553835a063db838c7489b66b897
#> #define SELU _rxa5ea762553835a063db838c7489b66b898
#> #define d4softplus _rxa5ea762553835a063db838c7489b66b899
#> #define d3softplus _rxa5ea762553835a063db838c7489b66b8100
#> #define d2softplus _rxa5ea762553835a063db838c7489b66b8101
#> #define dsoftplus _rxa5ea762553835a063db838c7489b66b8102
#> #define softplus _rxa5ea762553835a063db838c7489b66b8103
#> #define d4GELU _rxa5ea762553835a063db838c7489b66b8104
#> #define d3GELU _rxa5ea762553835a063db838c7489b66b8105
#> #define d2GELU _rxa5ea762553835a063db838c7489b66b8106
#> #define dGELU _rxa5ea762553835a063db838c7489b66b8107
#> #define GELU _rxa5ea762553835a063db838c7489b66b8108
#> #define dReLU _rxa5ea762553835a063db838c7489b66b8109
#> #define ReLU _rxa5ea762553835a063db838c7489b66b8110
#> #define riweibull _rxa5ea762553835a063db838c7489b66b8111
#> #define riunif _rxa5ea762553835a063db838c7489b66b8112
#> #define rit_ _rxa5ea762553835a063db838c7489b66b8113
#> #define ripois _rxa5ea762553835a063db838c7489b66b8114
#> #define ribeta _rxa5ea762553835a063db838c7489b66b8115
#> #define rigamma _rxa5ea762553835a063db838c7489b66b8116
#> #define rigeom _rxa5ea762553835a063db838c7489b66b8117
#> #define rif _rxa5ea762553835a063db838c7489b66b8118
#> #define riexp _rxa5ea762553835a063db838c7489b66b8119
#> #define richisq _rxa5ea762553835a063db838c7489b66b8120
#> #define ricauchy _rxa5ea762553835a063db838c7489b66b8121
#> #define rinbinomMu _rxa5ea762553835a063db838c7489b66b8122
#> #define rinbinom _rxa5ea762553835a063db838c7489b66b8123
#> #define ribinom _rxa5ea762553835a063db838c7489b66b8124
#> #define rinorm _rxa5ea762553835a063db838c7489b66b8125
#> #define rxweibull _rxa5ea762553835a063db838c7489b66b8126
#> #define rxunif _rxa5ea762553835a063db838c7489b66b8127
#> #define rxt_ _rxa5ea762553835a063db838c7489b66b8128
#> #define rxpois _rxa5ea762553835a063db838c7489b66b8129
#> #define rxbeta _rxa5ea762553835a063db838c7489b66b8130
#> #define rxgamma _rxa5ea762553835a063db838c7489b66b8131
#> #define rxgeom _rxa5ea762553835a063db838c7489b66b8132
#> #define rxf _rxa5ea762553835a063db838c7489b66b8133
#> #define rxexp _rxa5ea762553835a063db838c7489b66b8134
#> #define rxchisq _rxa5ea762553835a063db838c7489b66b8135
#> #define rxcauchy _rxa5ea762553835a063db838c7489b66b8136
#> #define rxnbinomMu _rxa5ea762553835a063db838c7489b66b8137
#> #define rxnbinom _rxa5ea762553835a063db838c7489b66b8138
#> #define rxbinom _rxa5ea762553835a063db838c7489b66b8139
#> #define rxnorm _rxa5ea762553835a063db838c7489b66b8140
#> #define linCmtC _rxa5ea762553835a063db838c7489b66b8141
#> #define linCmtB _rxa5ea762553835a063db838c7489b66b8142
#> #define linCmtA _rxa5ea762553835a063db838c7489b66b8143
#> #include <rxode2_model_shared.h>
#> #define __MAX_PROD__ 0
#> #define _CMT CMT
#> #define _SYNC_simeps_ for (int _svari=_solveData->neps; _svari--;){  if (_solveData->svar[_svari] == 0) {lka = _PP[0];};   if (_solveData->svar[_svari] == 1) {lcl = _PP[1];};   if (_solveData->svar[_svari] == 2) {lvc = _PP[2];};   if (_solveData->svar[_svari] == 3) {propSd = _PP[3];};   if (_solveData->svar[_svari] == 4) {rxerr_DoT_Cc = _PP[4];}; }
#> #define _SYNC_simeta_ for (int _ovari=_solveData->neta; _ovari--;){  if (_solveData->ovar[_ovari] == 0) {lka = _PP[0];};   if (_solveData->ovar[_ovari] == 1) {lcl = _PP[1];};   if (_solveData->ovar[_ovari] == 2) {lvc = _PP[2];};   if (_solveData->ovar[_ovari] == 3) {propSd = _PP[3];};   if (_solveData->ovar[_ovari] == 4) {rxerr_DoT_Cc = _PP[4];}; }
#> _getRxSolve_t _getRxSolve_;
#> _simfun simeps;
#> _simfun simeta;
#> _udf_type _evalUdf=NULL;
#> rx_solve *_solveData=NULL;
#> rxode2_assign_ptr _assign_ptr=NULL;
#> _rxRmModelLibType _rxRmModelLib=NULL;
#> _rxGetModelLibType _rxGetModelLib=NULL;
#> rxode2_ode_solver_old_c _old_c=NULL;
#> rxode2_fn0i _ptrid=NULL;
#> _rxIsCurrentC_type _rxIsCurrentC=NULL;
#> _rxSumType _sumPS=NULL;
#> _rxProdType _prodPS=NULL;
#> rxode2_fn0i _prodType=NULL;
#> rxode2_fn0i _sumType=NULL;
#> _update_par_ptr_p _update_par_ptr=NULL;
#> _getParCov_p _getParCov=NULL;
#> linCmtA_p linCmtA;
#> linCmtA_p linCmtC;
#> linCmtB_p linCmtB;
#> _rx_asgn _rxode2_rxAssignPtr=NULL;
#> _rx_asgn _rxQr=NULL;
#> rxode2_fn phi;
#> rxode2_fn ReLU;
#> rxode2_fn dReLU;
#> rxode2_fn GELU;
#> rxode2_fn dGELU;
#> rxode2_fn d2GELU;
#> rxode2_fn d3GELU;
#> rxode2_fn d4GELU;
#> rxode2_fn softplus;
#> rxode2_fn dsoftplus;
#> rxode2_fn d2softplus;
#> rxode2_fn d3softplus;
#> rxode2_fn d4softplus;
#> rxode2_fn SELU;
#> rxode2_fn dSELU;
#> rxode2_fn lReLU;
#> rxode2_fn dlReLU;
#> rxode2_fn Swish;
#> rxode2_fn dSwish;
#> rxode2_fn2 PReLU;
#> rxode2_fn2 dPReLU;
#> rxode2_fn2 dPReLUa;
#> rxode2_fn2 dPReLUa1;
#> rxode2_fn2 ELU;
#> rxode2_fn2 dELU;
#> rxode2_fn2 d2ELU;
#> rxode2_fn2 d2aELU;
#> rxode2_fn2 dELUa;
#> rxode2_fn2 d2ELUa;
#> rxode2_fn3 logit;
#> rxode2_fn3 expit;
#> rxode2_fn2 gammap;
#> rxode2_fn2 gammaq;
#> rxode2_fn2 lowergamma;
#> rxode2_fn2 uppergamma;
#> rxode2_fn2 gammapInv;
#> rxode2_fn2 gammapDer;
#> rxode2_fn2 gammapInva;
#> rxode2_fn2 gammaqInv;
#> rxode2_fn2 gammaqInva;
#> rxode2i_fn2 rxnorm;
#> rxode2i_rxbinom rxbinom;
#> rxode2i_rxbinom rxnbinom;
#> rxode2i_rxbinom rxnbinomMu;
#> rxode2i_fn2 rxcauchy;
#> rxode2i_fn rxchisq;
#> rxode2i_fn rxexp;
#> rxode2i_fn2 rxf;
#> rxode2i_ifn rxgeom;
#> rxode2i_fn2 rxgamma;
#> rxode2i_fn2 rxbeta;
#> rxode2i_ifn rxpois;
#> rxode2i_fn rxt_;
#> rxode2i_fn2 rxunif;
#> rxode2i_fn2 rxweibull;
#> rxode2i2_fn2 rinorm;
#> rxode2i2_ribinom ribinom;
#> rxode2i2_ribinom rinbinom;
#> rxode2i2_ribinom rinbinomMu;
#> rxode2i2_fn2 ricauchy;
#> rxode2i2_fn richisq;
#> rxode2i2_fn riexp;
#> rxode2i2_fn2 rif;
#> rxode2i2_ifn rigeom;
#> rxode2i2_fn2 rigamma;
#> rxode2i2_fn2 ribeta;
#> rxode2i2_ifn ripois;
#> rxode2i2_fn rit_;
#> rxode2i2_fn2 riunif;
#> rxode2i2_fn2 riweibull;
#> rxode2_llikNormFun _llikNorm;
#> rxode2_llikNormFun _llikNormDmean;
#> rxode2_llikNormFun _llikNormDsd;
#> rxode2_llikPoisFun _llikPois;
#> rxode2_llikPoisFun _llikPoisDlambda;
#> rxode2_llikBinomFun _llikBinom;
#> rxode2_llikBinomFun _llikBinomDprob;
#> rxode2_llikBinomFun _llikNbinom;
#> rxode2_llikBinomFun _llikNbinomDprob;
#> rxode2_llikBinomFun _llikNbinomMu;
#> rxode2_llikBinomFun _llikNbinomMuDmu;
#> rxode2_llikBetaFun _llikBeta;
#> rxode2_llikBetaFun _llikBetaDshape1;
#> rxode2_llikBetaFun _llikBetaDshape2;
#> rxode2_llikTFun _llikT;
#> rxode2_llikTFun _llikTDdf;
#> rxode2_llikTFun _llikTDmean;
#> rxode2_llikTFun _llikTDsd;
#> rxode2_llikChisqFun _llikChisq;
#> rxode2_llikChisqFun _llikChisqDdf;
#> rxode2_llikExpFun _llikExp;
#> rxode2_llikExpFun _llikExpDrate;
#> rxode2_llikFFun _llikF;
#> rxode2_llikFFun _llikFDdf1;
#> rxode2_llikFFun _llikFDdf2;
#> rxode2_llikGeomFun _llikGeom;
#> rxode2_llikGeomFun _llikGeomDp;
#> rxode2_llikUnifFun _llikUnif;
#> rxode2_llikUnifFun _llikUnifDalpha;
#> rxode2_llikUnifFun _llikUnifDbeta;
#> rxode2_llikWeibullFun _llikWeibull;
#> rxode2_llikWeibullFun _llikWeibullDshape;
#> rxode2_llikWeibullFun _llikWeibullDscale;
#> rxode2_llikGammaFun _llikGamma;
#> rxode2_llikGammaFun _llikGammaDshape;
#> rxode2_llikGammaFun _llikGammaDrate;
#> rxode2_llikCauchyFun _llikCauchy;
#> rxode2_llikCauchyFun _llikCauchyDlocation;
#> rxode2_llikCauchyFun _llikCauchyDscale;
#> rxode2_compareFactorVal_fn _compareFactorVal;
#> #include "extraC.h"
#> double _prod(double *input, double *p, int type, int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   return _prodPS(input, p, n, type);
#> }
#> double _udf(const char *funName, double *input, int n, ...) {
#>   if (n == -42) Rf_error("%s", "this has a ui user function that cannot be called directly");
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   return _evalUdf(funName, n, input);
#> }
#> double _sum(double *input, double *pld, int m, int type, int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   for (unsigned int i = 0; i < n; i++){
#>     input[i] = va_arg(valist, double);
#>   }
#>   va_end(valist);
#>   double ret = _sumPS(input, n, pld, m, type);
#>   if (type == 2 && m < 0){
#>     for (int i = -m; i--;){
#>       pld[i] = 0.0;
#>     }
#>   }
#>   return ret;
#> }
#> double _sign(unsigned int n, ...) {
#>   va_list valist;
#>   va_start(valist, n);
#>   double s = 1;
#>   for (unsigned int i = 0; i < n; i++) {
#>     s = sign(va_arg(valist, double))*s;
#>     if (s == 0){
#>       break;
#>     }
#>   }
#>   va_end(valist);
#>   return s;
#> }
#> double _rxord(int _cSub, unsigned int n,  ...) {
#>   rx_solving_options_ind* ind = &(_solveData->subjects[_cSub]);
#>   if (!ind->inLhs) {
#>     return 1.0;
#>   }
#>   va_list valist;
#>   va_start(valist, n);
#>   double ret = 1.0;
#>   double p = 0.0;
#>   double u = rxunif(ind, 0.0, 1.0);
#>   int found = 0;
#>   for (unsigned int i = 0; i < n; i++) {
#>     p += va_arg(valist, double);
#>     if (!found) {
#>       if (u < p) {
#>         ret = (double)(i+1);
#>         found = 1;
#>       }
#>     }
#>   }
#>   if (!found) ret =(double)(n+1);
#>   va_end(valist);
#>   return ret;
#> }
#> double _max(unsigned int n, ...) {
#>   va_list valist;
#>   va_start(valist, n);
#>   double mx = NA_REAL;
#>   double tmp = 0;
#>   if (n >= 1){
#>     mx = va_arg(valist, double);
#>     for (unsigned int i = 1; i < n; i++) {
#>       tmp = va_arg(valist, double);
#>       if (tmp>mx) mx=tmp;
#>     }
#>     va_end(valist);
#>   }
#>   return mx;
#> }
#> double _min(unsigned int n, ...){
#>   va_list valist;
#>   va_start(valist, n);
#>   double mn = NA_REAL;
#>   double tmp = 0;
#>   if (n >= 1){
#>     mn = va_arg(valist, double);
#>     for (unsigned int i = 1; i < n; i++){
#>       tmp = va_arg(valist, double);
#>       if (tmp<mn) mn=tmp;
#>     }
#>     va_end(valist);
#>   }
#>   return mn;
#> }
#> double _transit4P(int cmt, double t, unsigned int id, double n, double mtt, double bio){
#>   double nd = (double) n;
#>   double ktr = (nd+1)/mtt;
#>   double lktr = _safe_log(nd+1)-_safe_log(mtt);
#>   double tlast = _solveData->subjects[id].tlastS[cmt];
#>   double dose = _solveData->subjects[id].curDoseS[cmt];
#>   if (ISNA(dose)) dose = 0.0;
#>   if (ISNA(tlast)) tlast = 0.0;
#>   double tad = (t-tlast);
#>   return exp(_safe_log(bio*dose)+lktr+n*(lktr+_safe_log(tad))-ktr*(tad)-lgamma1p(nd));
#> }
#> double _transit3P(int cmt, double t, unsigned int id, double n, double mtt){
#>   double nd = (double) n;
#>   double ktr = (nd+1)/mtt;
#>   double lktr = _safe_log(nd+1)-_safe_log(mtt);
#>   double tlast = _solveData->subjects[id].tlastS[cmt];
#>   if (ISNA(tlast)) tlast = 0.0;
#>   double tad = t-tlast;
#>   double podo = _solveData->subjects[id].curDoseS[cmt];
#>   if (ISNA(podo)) podo = 0.0;
#>   return exp(_safe_log(podo)+lktr+n*(lktr+_safe_log(tad))-ktr*(tad)-lgamma1p(nd));
#> }
#> void _assignFuns0(void) {
#>   _evalUdf = (_udf_type) R_GetCCallable("rxode2", "_rxode2_evalUdf");
#>   _getRxSolve_ = (_getRxSolve_t) R_GetCCallable("rxode2","getRxSolve_");
#>   _assign_ptr=(rxode2_assign_ptr) R_GetCCallable("rxode2","rxode2_assign_fn_pointers");
#>   _rxRmModelLib=(_rxRmModelLibType) R_GetCCallable("rxode2","rxRmModelLib");
#>   _rxGetModelLib=(_rxGetModelLibType) R_GetCCallable("rxode2","rxGetModelLib");
#>   _rxode2_rxAssignPtr=(_rx_asgn)R_GetCCallable("rxode2","_rxode2_rxAssignPtr");
#>   _rxQr=(_rx_asgn)R_GetCCallable("rxode2","_rxode2_rxQr");
#>   _rxIsCurrentC = (_rxIsCurrentC_type)R_GetCCallable("rxode2","rxIsCurrentC");
#>   _sumPS  = (_rxSumType) R_GetCCallable("PreciseSums","PreciseSums_sum_r");
#>   _prodPS = (_rxProdType) R_GetCCallable("PreciseSums","PreciseSums_prod_r");
#>   _prodType=(rxode2_fn0i)R_GetCCallable("PreciseSums", "PreciseSums_prod_get");
#>   _sumType=(rxode2_fn0i)R_GetCCallable("PreciseSums", "PreciseSums_sum_get");
#>   _ptrid=(rxode2_fn0i)R_GetCCallable("rxode2", "rxode2_current_fn_pointer_id");
#>   _compareFactorVal=(rxode2_compareFactorVal_fn) R_GetCCallable("rxode2", "compareFactorVal");
#>   _update_par_ptr = (_update_par_ptr_p) R_GetCCallable("rxode2","_update_par_ptr");
#>   _getParCov = (_getParCov_p) R_GetCCallable("rxode2","_getParCov");
#>   // dynamic start
#>   _llikCauchyDscale = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchyDscale");
#>   _llikCauchyDlocation = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchyDlocation");
#>   _llikCauchy = (rxode2_llikCauchyFun) R_GetCCallable("rxode2ll", "rxLlikCauchy");
#>   _llikGammaDrate = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGammaDrate");
#>   _llikGammaDshape = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGammaDshape");
#>   _llikGamma = (rxode2_llikGammaFun) R_GetCCallable("rxode2ll", "rxLlikGamma");
#>   _llikWeibullDscale = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibullDscale");
#>   _llikWeibullDshape = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibullDshape");
#>   _llikWeibull = (rxode2_llikWeibullFun) R_GetCCallable("rxode2ll", "rxLlikWeibull");
#>   _llikUnifDbeta = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnifDbeta");
#>   _llikUnifDalpha = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnifDalpha");
#>   _llikUnif = (rxode2_llikUnifFun) R_GetCCallable("rxode2ll", "rxLlikUnif");
#>   _llikGeomDp = (rxode2_llikGeomFun) R_GetCCallable("rxode2ll", "rxLlikGeomDp");
#>   _llikGeom = (rxode2_llikGeomFun) R_GetCCallable("rxode2ll", "rxLlikGeom");
#>   _llikFDdf2 = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikFDdf2");
#>   _llikFDdf1 = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikFDdf1");
#>   _llikF = (rxode2_llikFFun) R_GetCCallable("rxode2ll", "rxLlikF");
#>   _llikExpDrate = (rxode2_llikExpFun) R_GetCCallable("rxode2ll", "rxLlikExpDrate");
#>   _llikExp = (rxode2_llikExpFun) R_GetCCallable("rxode2ll", "rxLlikExp");
#>   _llikChisqDdf = (rxode2_llikChisqFun) R_GetCCallable("rxode2ll", "rxLlikChisqDdf");
#>   _llikChisq = (rxode2_llikChisqFun) R_GetCCallable("rxode2ll", "rxLlikChisq");
#>   _llikTDsd = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDsd");
#>   _llikTDmean = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDmean");
#>   _llikTDdf = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikTDdf");
#>   _llikT = (rxode2_llikTFun) R_GetCCallable("rxode2ll", "rxLlikT");
#>   _llikBetaDshape2 = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBetaDshape2");
#>   _llikBetaDshape1 = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBetaDshape1");
#>   _llikBeta = (rxode2_llikBetaFun) R_GetCCallable("rxode2ll", "rxLlikBeta");
#>   _llikNbinomMuDmu = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomMuDmu");
#>   _llikNbinomMu = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomMu");
#>   _llikNbinomDprob = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinomDprob");
#>   _llikNbinom = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikNbinom");
#>   _llikBinomDprob = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikBinomDprob");
#>   _llikBinom = (rxode2_llikBinomFun) R_GetCCallable("rxode2ll", "rxLlikBinom");
#>   _llikPoisDlambda = (rxode2_llikPoisFun) R_GetCCallable("rxode2ll", "rxLlikPoisDlambda");
#>   _llikPois = (rxode2_llikPoisFun) R_GetCCallable("rxode2ll", "rxLlikPois");
#>   _llikNormDsd = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNormDsd");
#>   _llikNormDmean = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNormDmean");
#>   _llikNorm = (rxode2_llikNormFun) R_GetCCallable("rxode2ll", "rxLlikNorm");
#>   simeps = (_simfun) R_GetCCallable("rxode2", "simeps");
#>   simeta = (_simfun) R_GetCCallable("rxode2", "simeta");
#>   expit = (rxode2_fn3) R_GetCCallable("rxode2", "expit");
#>   logit = (rxode2_fn3) R_GetCCallable("rxode2", "logit");
#>   gammapDer = (rxode2_fn2) R_GetCCallable("rxode2", "gammapDer");
#>   lowergamma = (rxode2_fn2) R_GetCCallable("rxode2", "lowergamma");
#>   uppergamma = (rxode2_fn2) R_GetCCallable("rxode2", "uppergamma");
#>   gammaqInva = (rxode2_fn2) R_GetCCallable("rxode2", "gammaqInva");
#>   gammaqInv = (rxode2_fn2) R_GetCCallable("rxode2", "gammaqInv");
#>   gammapInva = (rxode2_fn2) R_GetCCallable("rxode2", "gammapInva");
#>   gammapInv = (rxode2_fn2) R_GetCCallable("rxode2", "gammapInv");
#>   gammaq = (rxode2_fn2) R_GetCCallable("rxode2", "gammaq");
#>   gammap = (rxode2_fn2) R_GetCCallable("rxode2", "gammap");
#>   phi = (rxode2_fn) R_GetCCallable("rxode2", "phi");
#>   d2ELUa = (rxode2_fn2) R_GetCCallable("rxode2", "d2ELUa");
#>   dELUa = (rxode2_fn2) R_GetCCallable("rxode2", "dELUa");
#>   d2aELU = (rxode2_fn2) R_GetCCallable("rxode2", "d2aELU");
#>   d2ELU = (rxode2_fn2) R_GetCCallable("rxode2", "d2ELU");
#>   dELU = (rxode2_fn2) R_GetCCallable("rxode2", "dELU");
#>   ELU = (rxode2_fn2) R_GetCCallable("rxode2", "ELU");
#>   dPReLUa1 = (rxode2_fn2) R_GetCCallable("rxode2", "dPReLUa1");
#>   dPReLUa = (rxode2_fn2) R_GetCCallable("rxode2", "dPReLUa");
#>   dPReLU = (rxode2_fn2) R_GetCCallable("rxode2", "dPReLU");
#>   PReLU = (rxode2_fn2) R_GetCCallable("rxode2", "PReLU");
#>   dSwish = (rxode2_fn) R_GetCCallable("rxode2", "dSwish");
#>   Swish = (rxode2_fn) R_GetCCallable("rxode2", "Swish");
#>   dlReLU = (rxode2_fn) R_GetCCallable("rxode2", "dlReLU");
#>   lReLU = (rxode2_fn) R_GetCCallable("rxode2", "lReLU");
#>   dSELU = (rxode2_fn) R_GetCCallable("rxode2", "dSELU");
#>   SELU = (rxode2_fn) R_GetCCallable("rxode2", "SELU");
#>   d4softplus = (rxode2_fn) R_GetCCallable("rxode2", "d4softplus");
#>   d3softplus = (rxode2_fn) R_GetCCallable("rxode2", "d3softplus");
#>   d2softplus = (rxode2_fn) R_GetCCallable("rxode2", "d2softplus");
#>   dsoftplus = (rxode2_fn) R_GetCCallable("rxode2", "dsoftplus");
#>   softplus = (rxode2_fn) R_GetCCallable("rxode2", "softplus");
#>   d4GELU = (rxode2_fn) R_GetCCallable("rxode2", "d4GELU");
#>   d3GELU = (rxode2_fn) R_GetCCallable("rxode2", "d3GELU");
#>   d2GELU = (rxode2_fn) R_GetCCallable("rxode2", "d2GELU");
#>   dGELU = (rxode2_fn) R_GetCCallable("rxode2", "dGELU");
#>   GELU = (rxode2_fn) R_GetCCallable("rxode2", "GELU");
#>   dReLU = (rxode2_fn) R_GetCCallable("rxode2", "dReLU");
#>   ReLU = (rxode2_fn) R_GetCCallable("rxode2", "ReLU");
#>   riweibull = (rxode2i2_fn2) R_GetCCallable("rxode2", "riweibull");
#>   riunif = (rxode2i2_fn2) R_GetCCallable("rxode2", "riunif");
#>   rit_ = (rxode2i2_fn) R_GetCCallable("rxode2", "rit_");
#>   ripois = (rxode2i2_ifn) R_GetCCallable("rxode2", "ripois");
#>   ribeta = (rxode2i2_fn2) R_GetCCallable("rxode2", "ribeta");
#>   rigamma = (rxode2i2_fn2) R_GetCCallable("rxode2", "rigamma");
#>   rigeom = (rxode2i2_ifn) R_GetCCallable("rxode2", "rigeom");
#>   rif = (rxode2i2_fn2) R_GetCCallable("rxode2", "rif");
#>   riexp = (rxode2i2_fn) R_GetCCallable("rxode2", "riexp");
#>   richisq = (rxode2i2_fn) R_GetCCallable("rxode2", "richisq");
#>   ricauchy = (rxode2i2_fn2) R_GetCCallable("rxode2", "ricauchy");
#>   rinbinomMu = (rxode2i2_ribinom) R_GetCCallable("rxode2", "rinbinomMu");
#>   rinbinom = (rxode2i2_ribinom) R_GetCCallable("rxode2", "rinbinom");
#>   ribinom = (rxode2i2_ribinom) R_GetCCallable("rxode2", "ribinom");
#>   rinorm = (rxode2i2_fn2) R_GetCCallable("rxode2", "rinorm");
#>   rxweibull = (rxode2i_fn2) R_GetCCallable("rxode2", "rxweibull");
#>   rxunif = (rxode2i_fn2) R_GetCCallable("rxode2", "rxunif");
#>   rxt_ = (rxode2i_fn) R_GetCCallable("rxode2", "rxt_");
#>   rxpois = (rxode2i_ifn) R_GetCCallable("rxode2", "rxpois");
#>   rxbeta = (rxode2i_fn2) R_GetCCallable("rxode2", "rxbeta");
#>   rxgamma = (rxode2i_fn2) R_GetCCallable("rxode2", "rxgamma");
#>   rxgeom = (rxode2i_ifn) R_GetCCallable("rxode2", "rxgeom");
#>   rxf = (rxode2i_fn2) R_GetCCallable("rxode2", "rxf");
#>   rxexp = (rxode2i_fn) R_GetCCallable("rxode2", "rxexp");
#>   rxchisq = (rxode2i_fn) R_GetCCallable("rxode2", "rxchisq");
#>   rxcauchy = (rxode2i_fn2) R_GetCCallable("rxode2", "rxcauchy");
#>   rxnbinomMu = (rxode2i_rxbinom) R_GetCCallable("rxode2", "rxnbinomMu");
#>   rxnbinom = (rxode2i_rxbinom) R_GetCCallable("rxode2", "rxnbinom");
#>   rxbinom = (rxode2i_rxbinom) R_GetCCallable("rxode2", "rxbinom");
#>   rxnorm = (rxode2i_fn2) R_GetCCallable("rxode2", "rxnorm");
#>   linCmtC = (linCmtA_p) R_GetCCallable("rxode2", "linCmtC");
#>   linCmtB = (linCmtB_p) R_GetCCallable("rxode2", "linCmtB");
#>   linCmtA = (linCmtA_p) R_GetCCallable("rxode2", "linCmtA");
#>   // dynamic stop
#>   _solveData = _getRxSolve_();
#> }
#> void _assignFuns(void) {
#>   if (_assign_ptr == NULL){
#>     _assignFuns0();
#>   }
#> }
#> void __assignFuns2(rx_solve rx,
#>                    rx_solving_options op,
#>                    t_F f,
#>                    t_LAG lag,
#>                    t_RATE rate,
#>                    t_DUR dur,
#>                    t_calc_mtime mtime,
#>                    t_ME me,
#>                    t_IndF indf,
#>                    t_getTime gettime,
#>                    t_locateTimeIndex timeindex,
#>                    t_handle_evidL handleEvid,
#>                    t_getDur getdur) {
#>   // assign start
#>     static rxode2_assignFuns2_t rxode2_assignFuns2 = NULL;
#>     if (rxode2_assignFuns2 == NULL) rxode2_assignFuns2 = (rxode2_assignFuns2_t)(R_GetCCallable("rxode2", "_rxode2_assignFuns2"));
#>     rxode2_assignFuns2(rx, op, f, lag, rate, dur, mtime, me, indf, gettime, timeindex, handleEvid, getdur);
#>   // assign stop
#> }
#> extern void  rx_3ad488a46759958958c15fcf7a453770__ode_solver_solvedata (rx_solve *solve){
#>   _solveData = solve;
#> }
#> extern rx_solve *rx_3ad488a46759958958c15fcf7a453770__ode_solver_get_solvedata(void){
#>   return _solveData;
#> }
#> SEXP rx_3ad488a46759958958c15fcf7a453770__model_vars(void);
#> 
#> 
#> // prj-specific differential eqns
#> void rx_3ad488a46759958958c15fcf7a453770__dydt(int *_neq, double __t, double *__zzStateVar__, double *__DDtStateVar__)
#> {
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   int _cSub = _neq[1];
#>   double t = __t + _solveData->subjects[_neq[1]].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=1;
#>     double lka = NA_REAL;
#>   double lcl = NA_REAL;
#>   double lvc = NA_REAL;
#>   double propSd = NA_REAL;
#>   double ka = NA_REAL;
#>   double cl = NA_REAL;
#>   double vc = NA_REAL;
#>   double kel = NA_REAL;
#>   double ka_t = NA_REAL;
#>   double depot = NA_REAL;
#>   double central = NA_REAL;
#>   double Cc = NA_REAL;
#>   double rx_yj___ = NA_REAL;
#>   double rx_lambda___ = NA_REAL;
#>   double rx_low___ = NA_REAL;
#>   double rx_hi___ = NA_REAL;
#>   double rx_pred_f_ = NA_REAL;
#>   double rx_pred_ = NA_REAL;
#>   double rx_r_ = NA_REAL;
#>   double ipredSim = NA_REAL;
#>   double sim = NA_REAL;
#>   double rxerr_DoT_Cc = NA_REAL;
#> 
#>   (void)t;
#>   (void)lka;
#>   (void)lcl;
#>   (void)lvc;
#>   (void)propSd;
#>   (void)ka;
#>   (void)cl;
#>   (void)vc;
#>   (void)kel;
#>   (void)ka_t;
#>   (void)depot;
#>   (void)central;
#>   (void)Cc;
#>   (void)rx_yj___;
#>   (void)rx_lambda___;
#>   (void)rx_low___;
#>   (void)rx_hi___;
#>   (void)rx_pred_f_;
#>   (void)rx_pred_;
#>   (void)rx_r_;
#>   (void)ipredSim;
#>   (void)sim;
#>   (void)rxerr_DoT_Cc;
#> 
#>   ka = _PL[0];
#>   cl = _PL[1];
#>   vc = _PL[2];
#>   kel = _PL[3];
#>   ka_t = _PL[4];
#>   Cc = _PL[5];
#>   ipredSim = _PL[6];
#>   sim = _PL[7];
#> 
#>   _update_par_ptr(__t, _cSub, _solveData, _idx);
#>   lka = _PP[0];
#>   lcl = _PP[1];
#>   lvc = _PP[2];
#>   propSd = _PP[3];
#>   rxerr_DoT_Cc = _PP[4];
#> 
#>   depot = __zzStateVar__[__DDT0__]*((double)(_ON[__DDT0__]));
#>   central = __zzStateVar__[__DDT1__]*((double)(_ON[__DDT1__]));
#> 
#>   ka =exp(lka);
#>   cl =exp(lcl);
#>   vc =exp(lvc);
#>   kel =cl/_div0(vc);
#>   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>   __DDtStateVar__[__DDT0__] = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] -ka_t*depot);
#>   __DDtStateVar__[__DDT1__] = ((double)(_ON[__DDT1__]))*(_IR[__DDT1__] + ka_t*depot-kel*central);
#>   Cc =central/_div0(vc);
#>   rx_yj_ =2;
#>   rx_lambda_ =1;
#>   rx_low_ =0;
#>   rx_hi_ =1;
#>   rx_pred_f_ =Cc;
#>   rx_pred_ =rx_pred_f_;
#>   rx_r_ =Rx_pow((rx_pred_f_*propSd),2);
#>   ipredSim =rxTBSi(rx_pred_,rx_lambda_,rx_yj_,rx_low_,rx_hi_);
#>   sim =rxTBSi(rx_pred_+sqrt(rx_r_)*rxerr_DoT_Cc,rx_lambda_,rx_yj_,rx_low_,rx_hi_);
#>   (&_solveData->subjects[_cSub])->dadt_counter[0]++;
#> }
#> 
#> // Jacobian derived vars
#> void rx_3ad488a46759958958c15fcf7a453770__calc_jac(int *_neq, double __t, double *__zzStateVar__, double *__PDStateVar__, unsigned int __NROWPD__) {
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>     int _cSub=_neq[1];
#>   double t = __t + _solveData->subjects[_neq[1]].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=2;
#>     (&_solveData->subjects[_cSub])->jac_counter[0]++;
#> }
#> // Functional based initial conditions.
#> void rx_3ad488a46759958958c15fcf7a453770__inis(int _cSub, double *__zzStateVar__){
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   
#>   (&_solveData->subjects[_cSub])->_rxFlag=3;
#>   }
#> // prj-specific derived vars
#> void rx_3ad488a46759958958c15fcf7a453770__calc_lhs(int _cSub, double __t, double *__zzStateVar__, double *_lhs) {
#>     int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=11;
#>     double  __DDtStateVar_0__;
#>   double  __DDtStateVar_1__;
#>   double  __DDtStateVar_2__;
#>   double lka = NA_REAL;
#>   double lcl = NA_REAL;
#>   double lvc = NA_REAL;
#>   double propSd = NA_REAL;
#>   double ka = NA_REAL;
#>   double cl = NA_REAL;
#>   double vc = NA_REAL;
#>   double kel = NA_REAL;
#>   double ka_t = NA_REAL;
#>   double depot = NA_REAL;
#>   double central = NA_REAL;
#>   double Cc = NA_REAL;
#>   double rx_yj___ = NA_REAL;
#>   double rx_lambda___ = NA_REAL;
#>   double rx_low___ = NA_REAL;
#>   double rx_hi___ = NA_REAL;
#>   double rx_pred_f_ = NA_REAL;
#>   double rx_pred_ = NA_REAL;
#>   double rx_r_ = NA_REAL;
#>   double ipredSim = NA_REAL;
#>   double sim = NA_REAL;
#>   double rxerr_DoT_Cc = NA_REAL;
#> 
#>   (void)t;
#>   (void)__DDtStateVar_0__;
#>   (void)__DDtStateVar_1__;
#>   (void)__DDtStateVar_2__;
#>   (void)lka;
#>   (void)lcl;
#>   (void)lvc;
#>   (void)propSd;
#>   (void)ka;
#>   (void)cl;
#>   (void)vc;
#>   (void)kel;
#>   (void)ka_t;
#>   (void)depot;
#>   (void)central;
#>   (void)Cc;
#>   (void)rx_yj___;
#>   (void)rx_lambda___;
#>   (void)rx_low___;
#>   (void)rx_hi___;
#>   (void)rx_pred_f_;
#>   (void)rx_pred_;
#>   (void)rx_r_;
#>   (void)ipredSim;
#>   (void)sim;
#>   (void)rxerr_DoT_Cc;
#> 
#>   ka = _PL[0];
#>   cl = _PL[1];
#>   vc = _PL[2];
#>   kel = _PL[3];
#>   ka_t = _PL[4];
#>   Cc = _PL[5];
#>   ipredSim = _PL[6];
#>   sim = _PL[7];
#> 
#>   _update_par_ptr(__t, _cSub, _solveData, _idx);
#>   lka = _PP[0];
#>   lcl = _PP[1];
#>   lvc = _PP[2];
#>   propSd = _PP[3];
#>   rxerr_DoT_Cc = _PP[4];
#> 
#>   depot = __zzStateVar__[__DDT0__]*((double)(_ON[__DDT0__]));
#>   central = __zzStateVar__[__DDT1__]*((double)(_ON[__DDT1__]));
#> 
#>   ka =exp(lka);
#>   cl =exp(lcl);
#>   vc =exp(lvc);
#>   kel =cl/_div0(vc);
#>   ka_t =ka*(1+exp(-0.1*_tad01(_DEPOT_)));
#>   __DDtStateVar_0__ = ((double)(_ON[__DDT0__]))*(_IR[__DDT0__] -ka_t*depot);
#>   __DDtStateVar_1__ = ((double)(_ON[__DDT1__]))*(_IR[__DDT1__] + ka_t*depot-kel*central);
#>   Cc =central/_div0(vc);
#>   rx_yj_ =2;
#>   rx_lambda_ =1;
#>   rx_low_ =0;
#>   rx_hi_ =1;
#>   rx_pred_f_ =Cc;
#>   rx_pred_ =rx_pred_f_;
#>   rx_r_ =Rx_pow((rx_pred_f_*propSd),2);
#>   ipredSim =rxTBSi(rx_pred_,rx_lambda_,rx_yj_,rx_low_,rx_hi_);
#>   sim =rxTBSi(rx_pred_+sqrt(rx_r_)*rxerr_DoT_Cc,rx_lambda_,rx_yj_,rx_low_,rx_hi_);
#> 
#>   _lhs[0]=ka;
#>   _lhs[1]=cl;
#>   _lhs[2]=vc;
#>   _lhs[3]=kel;
#>   _lhs[4]=ka_t;
#>   _lhs[5]=Cc;
#>   _lhs[6]=ipredSim;
#>   _lhs[7]=sim;
#> }
#> // Functional based bioavailability
#> double rx_3ad488a46759958958c15fcf7a453770__F(int _cSub,  int _cmt, double _amt, double __t, double *__zzStateVar__){
#>  return _amt;
#>   }
#> // Functional based absorption lag
#> double rx_3ad488a46759958958c15fcf7a453770__Lag(int _cSub,  int _cmt, double __t, double *__zzStateVar__){
#>  return __t;
#> }
#> // Modeled zero-order rate
#> double rx_3ad488a46759958958c15fcf7a453770__Rate(int _cSub,  int _cmt, double _amt, double __t, double *__zzStateVar__){
#>  return 0.0;
#> }
#> // Modeled zero-order duration
#> double rx_3ad488a46759958958c15fcf7a453770__Dur(int _cSub,  int _cmt, double _amt, double __t){
#>  return 0.0;
#> }
#> // Model Times
#> void rx_3ad488a46759958958c15fcf7a453770__mtime(int _cSub, double *_mtime){
#> }
#> // Matrix Exponential (0)
#> void rx_3ad488a46759958958c15fcf7a453770__ME(int _cSub, double _t, double __t, double *_mat, const double *__zzStateVar__){
#>   int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=9;
#>   }
#> // Inductive linearization Matf
#> void rx_3ad488a46759958958c15fcf7a453770__IndF(int _cSub, double _t, double __t, double *_matf){
#>  int _itwhile = 0;
#>   (void)_itwhile;
#>   double t = __t + _solveData->subjects[_cSub].curShift;
#>   (void)t;
#>   (&_solveData->subjects[_cSub])->_rxFlag=10;
#>   }
#> extern SEXP rx_3ad488a46759958958c15fcf7a453770__model_vars(void){
#>   int pro=0;
#>   SEXP _mv = PROTECT(_rxGetModelLib("rx_3ad488a46759958958c15fcf7a453770__model_vars"));pro++;
#>   if (!_rxIsCurrentC(_mv)){
#>     SEXP hash    = PROTECT(Rf_allocVector(STRSXP, 1));pro++;
#> #define __doBuf__  snprintf(buf, __doBufN__, "un]\"BAAA@QRtHACAAAAAAAjn.AAAv7#aT)\"2!{RAd9Bi@(%%xaBKCe4ouXEEcEi3XoX$^ZHbPeqS}_cOM@cYO|X~cqO&Vr.vlx$,k}7vEmIGWX!B*/+D1$E0Bbe4QgGrHg$;vFAizWJ0dA^I;fhu^m.BN/=oCx|{9sH*Wu,c{KS5$I[n/%%:KPk7gTT]F~:Pk{IP,lAp%%vasXKaPu#Q4^x4`udU:2@I.Cs46Lm4FQO5$)z4;.\?#udinw69<*=6z6VCIx)`+4=^_1SbPRxr|*md7V!8}0Y936sm6kuNj!$bc}\?sGs.vuEO`RGFOWElKSo~PoRuEBj4qYv@:}V&~&OROJFdMD[;CHT}T!:~XP)a_}0$,fU:oeSy2E4bO<\?*Tc!,WqC\"R%%y4Po4]``wYGRE6f#2A4Zx<F*ou_$>v/;E*r\"RC#h+&<zs^(r6:tab(4Om%%jP!#:]#Vc^Zuir1(~L#n.{OC,@;m/{CZbO4uD6NefeoTLHT6N;4v@(s#~y!^.WosCk_f7,BCM+8YcOceVSEtHaLF_ywZ,06>VC8.EHsGHd4Mjy489Dgt4AIDDqPw=*KK8<8rW:Z\"77k^\?`&8mA@;V]s\"lIr@H,Q}V|#iIACP*)9I$^U+{u^VyLxZa4*G&E{PE]9^eKcc}5\?]IAs<Mozwc_MaZ:y%%Z,>m3>OBV8a/MzoOV0]pJi@rIcQ^r$[Abi3T26d92Lh}:Xec2X15DsiU6Myto/0O:MW!@B3VtQwy;VaqB]+NQ9(ym96a*F,1Dm^fM>hrUkicMjrxw9Qt=e\"8{NOvM^P7/GC2+\"13!v3<P\?9h+#U<1qT$&>cWO,}%%0p|!0U+|4*w!<9^%%`pfbilh=qWC~8$IJ_*:aGKj]1^\?:gXu(5$PY6Fl_.xe;{rm~wRaA=7fQ#*:,/<98mcE%%z$FXE9x5aY:Y[!5RRl&Y[;]1f>+[lyvo@Qz2D~1CChBi6Vcg5xIEj/xKZ):.iV[og^2S(4{)C=)@O{B>9^:C\"SY<T7wFUQ2!BpL>MozrCxB\?v9E_,;AK2rVX@]CUX=v9F2B6I*MADl!2[>T\?(x#BVy\?bU*~Y.JX~D7Rf\"53>V+[$)JHE&3zG_|]aRABNuhdk5~^{2FCh%%o=hxDbl~bRBJG&aN#)HTt*SGyP3Tj4)JOoB1D@^)(a)~;d;S]kQD&|Nqhql}M>4<w1cb4}EUti2H=eBY^l\"YrNNwK2]*$U}`ZErnvOZGXEv|wv)^Y3PeC!vtX<90[;d2Dcky>\"`Q&]:kehwooBKeZO,60pVcHe9,w/EIZM9s~:3=\":{w:7oE5.Yh^wh$.mWM#Eg<`tuCT5Rk}0X5p^X+!GHYn{|b1H<w0^k)4d(,WrjuChs#=luWP\?FI/%%BA$Fnj~TL1_UAu7DuLMQf[zX1}>BsVcwSOOrHMzKI;\"_WBl(Mcl@bKV\?i\")z9E2cK6]_#ZY=BNp=lb~7w5pc6euIR68biV05");
#>     char buf[1470];
#> #define __doBufN__ 1470
#>     __doBuf__
#> #undef __doBuf__
#> #undef __doBufN__
#>     SET_STRING_ELT(hash, 0, mkChar(buf));
#>     SEXP lst      = PROTECT(_rxQr(hash));pro++;
#>     _assign_ptr(lst);
#>     UNPROTECT(pro);
#>     return lst;
#>   } else {
#>     UNPROTECT(pro);
#>     return _mv;
#>   }
#> }
#> extern void rx_3ad488a46759958958c15fcf7a453770__dydt_lsoda(int *neq, double *t, double *A, double *DADT)
#> {
#>   rx_3ad488a46759958958c15fcf7a453770__dydt(neq, *t, A, DADT);
#> }
#> extern int rx_3ad488a46759958958c15fcf7a453770__dydt_liblsoda(double __t, double *y, double *ydot, void *data)
#> {
#>   int *neq = (int*)(data);
#>   rx_3ad488a46759958958c15fcf7a453770__dydt(neq, __t, y, ydot);
#>   return(0);
#> }
#> extern void rx_3ad488a46759958958c15fcf7a453770__calc_jac_lsoda(int *neq, double *t, double *A,int *ml, int *mu, double *JAC, int *nrowpd){
#>   // Update all covariate parameters
#>   rx_3ad488a46759958958c15fcf7a453770__calc_jac(neq, *t, A, JAC, *nrowpd);
#> }
#> 
#> //Create function to call from R's main thread that assigns the required functions. Sometimes they don't get assigned.
#> extern void rx_3ad488a46759958958c15fcf7a453770__assignFuns(void){
#>   _assignFuns();
#> }
#> 
#> //Initialize the dll to match rxode2's calls
#> void R_init0_rx_3ad488a46759958958c15fcf7a453770_(void){
#>   // Get C callables on load; Otherwise it isn't thread safe
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__assignFuns2", (DL_FUNC) __assignFuns2);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__assignFuns", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__assignFuns);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__inis",(DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__inis);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__dydt",(DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__dydt);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__calc_lhs",(DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__calc_lhs);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__calc_jac",(DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__calc_jac);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__dydt_lsoda", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__dydt_lsoda);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__calc_jac_lsoda", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__calc_jac_lsoda);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__ode_solver_solvedata", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__ode_solver_solvedata);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__ode_solver_get_solvedata", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__ode_solver_get_solvedata);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__F", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__F);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__Lag", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__Lag);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__Rate", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__Rate);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__Dur", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__Dur);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__mtime", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__mtime);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__ME", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__ME);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__IndF", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__IndF);
#>   R_RegisterCCallable("rx_3ad488a46759958958c15fcf7a453770_","rx_3ad488a46759958958c15fcf7a453770__dydt_liblsoda", (DL_FUNC) rx_3ad488a46759958958c15fcf7a453770__dydt_liblsoda);
#> }
#> //Initialize the dll to match rxode2's calls
#> void R_init_rx_3ad488a46759958958c15fcf7a453770_(DllInfo *info){
#>   // Get C callables on load; Otherwise it isn't thread safe
#>   R_init0_rx_3ad488a46759958958c15fcf7a453770_();
#>   static const R_CallMethodDef callMethods[]  = {
#>     {"rx_3ad488a46759958958c15fcf7a453770__model_vars", (DL_FUNC) &rx_3ad488a46759958958c15fcf7a453770__model_vars, 0},
#>     {NULL, NULL, 0}
#>   };
#> 
#>   R_registerRoutines(info, NULL, callMethods, NULL, NULL);
#>   R_useDynamicSymbols(info,FALSE);
#>   _assignFuns0();
#> 
#> }
#> 
#> void R_unload_rx_3ad488a46759958958c15fcf7a453770_ (DllInfo *info){
#>   // Free resources required for single subject solve.
#>   SEXP _mv = PROTECT(_rxGetModelLib("rx_3ad488a46759958958c15fcf7a453770__model_vars"));
#>   if (!isNull(_mv)){
#>     _rxRmModelLib("rx_3ad488a46759958958c15fcf7a453770__model_vars");
#>   }
#>   UNPROTECT(1);
#> }
#> Error: error building model
```

<sup>Created on 2024-11-30 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,5 +1,9 @@
 # rxode2 (development version)
 
+- Rework the `tad()` and related functions so they use the same
+  interface as compartments (this way they do not depend on the order
+  of compartments); See #815
+
 - Add `getRxNpars` to api.  This allows the development version of
   `babelmixr2` to better check what model is loaded and unload/reload
   as necessary.

--- a/NEWS.md
+++ b/NEWS.md
@@ -2,7 +2,8 @@
 
 - Rework the `tad()` and related functions so they use the same
   interface as compartments (this way they do not depend on the order
-  of compartments); See #815
+  of compartments); See #815.  Also allow dummy variables to ignore
+  state requirements (for parsing mu-referenced equations).
 
 - Add `getRxNpars` to api.  This allows the development version of
   `babelmixr2` to better check what model is loaded and unload/reload

--- a/src/codegen.c
+++ b/src/codegen.c
@@ -676,8 +676,8 @@ SEXP _rxode2_codegen(SEXP c_file, SEXP prefix, SEXP libname,
     if (tb.hasKa == 1) {
       sAppend(&sbOut, "#define _DEPOT_ %d\n", tb.statei);
       sAppend(&sbOut, "#define _CENTRAL_ %d\n", tb.statei+1);
-    } else if (tb.hasCentral == 1) {
-      if (tb.hasDepot){
+    } else if (tb.hasCentralCmt == 1) {
+      if (tb.hasDepotCmt){
         fclose(fpIO);
         _rxode2parse_unprotect();
         err_trans("linCmt() does not have 'depot' compartment without a 'ka'");

--- a/src/genModelVars.h
+++ b/src/genModelVars.h
@@ -218,6 +218,36 @@ static inline int sortStateVectorsErrHandle(int prop, int pass, int i) {
   if ((prop & propDur) != 0) {
     sAppend(&sbt, "'dur(%s)', ", tb.ss.line[tb.di[i]]);
   }
+  if ((prop & propTad) != 0) {
+    sAppend(&sbt, "'tad(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propTad0) != 0) {
+    sAppend(&sbt, "'tad0(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propTafd) != 0) {
+    sAppend(&sbt, "'tafd(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propTafd0) != 0) {
+    sAppend(&sbt, "'tafd0(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propTlast) != 0) {
+    sAppend(&sbt, "'tlast(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propTlast0) != 0) {
+    sAppend(&sbt, "'tlast0(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propTfirst) != 0) {
+    sAppend(&sbt, "'tfirst(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propTfirst0) != 0) {
+    sAppend(&sbt, "'tfirst0(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propPodo) != 0) {
+    sAppend(&sbt, "'podo(%s)', ", tb.ss.line[tb.di[i]]);
+  }
+  if ((prop & propDose) != 0) {
+    sAppend(&sbt, "'dose(%s)', ", tb.ss.line[tb.di[i]]);
+  }
   // Take off trailing "',
   sbt.o -= 2;
   sbt.s[sbt.o] = 0;

--- a/src/genModelVars.h
+++ b/src/genModelVars.h
@@ -105,13 +105,13 @@ static inline void calcNparamsNlhsNslhs(void) {
 
 static inline void calcNextra(void) {
   int offCmt=0,nExtra = 0;
-  char *buf, buf2[200];
+  char *buf=NULL, buf2[200];
   for (int i = 0; i < tb.statei; i++){
     if (offCmt == 0 && tb.idu[i] == 0){
+      buf=tb.ss.line[tb.di[i]];
       offCmt = 1;
       nExtra++;
-      buf=tb.ss.line[tb.di[i]];
-    } else if (offCmt == 1 && tb.idu[i] == 1){
+    } else if (offCmt == 1 && tb.idu[i] == 1) {
       // There is an compartment that doesn't have a derivative
       if (tb.linCmt == 0){
         char *v = rc_dup_str(buf, 0);
@@ -203,55 +203,57 @@ static inline int sortStateVectorsErrHandle(int prop, int pass, int i) {
   if (prop == 0 || pass == 1) {
     return 1;
   }
+  char *buf = NULL;
+  buf = tb.ss.line[tb.di[i]];
   if ((prop & prop0) != 0) {
-    sAppend(&sbt, "'%s(0)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'%s(0)', ", buf);
   }
   if ((prop & propF) != 0) {
-    sAppend(&sbt, "'f(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'f(%s)', ", buf);
   }
   if ((prop & propAlag) != 0) {
-    sAppend(&sbt, "'alag(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'alag(%s)', ", buf);
   }
   if ((prop & propRate) != 0) {
-    sAppend(&sbt, "'rate(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'rate(%s)', ", buf);
   }
   if ((prop & propDur) != 0) {
-    sAppend(&sbt, "'dur(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'dur(%s)', ", buf);
   }
   if ((prop & propTad) != 0) {
-    sAppend(&sbt, "'tad(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tad(%s)', ", buf);
   }
   if ((prop & propTad0) != 0) {
-    sAppend(&sbt, "'tad0(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tad0(%s)', ", buf);
   }
   if ((prop & propTafd) != 0) {
-    sAppend(&sbt, "'tafd(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tafd(%s)', ", buf);
   }
   if ((prop & propTafd0) != 0) {
-    sAppend(&sbt, "'tafd0(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tafd0(%s)', ", buf);
   }
   if ((prop & propTlast) != 0) {
-    sAppend(&sbt, "'tlast(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tlast(%s)', ", buf);
   }
   if ((prop & propTlast0) != 0) {
-    sAppend(&sbt, "'tlast0(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tlast0(%s)', ", buf);
   }
   if ((prop & propTfirst) != 0) {
-    sAppend(&sbt, "'tfirst(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tfirst(%s)', ", buf);
   }
   if ((prop & propTfirst0) != 0) {
-    sAppend(&sbt, "'tfirst0(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'tfirst0(%s)', ", buf);
   }
   if ((prop & propPodo) != 0) {
-    sAppend(&sbt, "'podo(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'podo(%s)', ", buf);
   }
   if ((prop & propDose) != 0) {
-    sAppend(&sbt, "'dose(%s)', ", tb.ss.line[tb.di[i]]);
+    sAppend(&sbt, "'dose(%s)', ", buf);
   }
   // Take off trailing "',
   sbt.o -= 2;
   sbt.s[sbt.o] = 0;
-  sAppend(&sbt, " present, but d/dt(%s) not defined\n", tb.ss.line[tb.di[i]]);
+  sAppend(&sbt, " present, but d/dt(%s) not defined\n", buf);
   return 0;
 }
 

--- a/src/genModelVars.h
+++ b/src/genModelVars.h
@@ -200,7 +200,7 @@ static inline SEXP calcIniVals(void) {
 SEXP orderForderS1(SEXP ordIn);
 
 static inline int sortStateVectorsErrHandle(int prop, int pass, int i) {
-  if (prop == 0 || pass == 1) {
+  if (prop == 0 || pass == 1 || tb.dummyLhs == 1) {
     return 1;
   }
   char *buf = NULL;

--- a/src/genModelVars.h
+++ b/src/genModelVars.h
@@ -268,7 +268,7 @@ static inline SEXP sortStateVectors(SEXP ordS) {
     int cur = tb.didx[i];
     int prop = tb.dprop[i];
     int pass = 0;
-    if (tb.linCmt){
+    if (tb.linCmt) {
       if (tb.hasDepotCmt == 1 && !strcmp("depot", tb.ss.line[tb.di[i]])){
         pass = 1;
       } else if ((tb.hasCentralCmt == 1 || tb.hasDepotCmt == 1)  &&

--- a/src/parseCmtProperties.h
+++ b/src/parseCmtProperties.h
@@ -331,7 +331,6 @@ static inline int handleRemainingAssignmentsCalcProps(nodeInfo ni, char *name, i
   return 0;
 }
 
-
 static inline int finalizeLineParam(nodeInfo ni, char *name) {
   if (nodeHas(param_statement)) {
     sbDt.o = 0; sbt.o = 0;

--- a/src/parseCmtProperties.h
+++ b/src/parseCmtProperties.h
@@ -183,6 +183,9 @@ static inline int handleRemainingAssignmentsCalcPropComplexAssign(nodeInfo ni, c
       // New assignment
       tb.ixL = tb.ix;
       tb.lh[tb.ix] = isLHS;
+      if (!strcmp(v, "rxdummyLhs")) {
+        tb.dummyLhs = 1;
+      }
     } else if (tb.ix < 0){
       if (!strcmp("rxlin___", v)) {
         tb.ixL=-1;
@@ -216,6 +219,9 @@ static inline int handleRemainingAssignmentsCalcPropComplexAssign(nodeInfo ni, c
         tb.lh[tb.ix] = isLHSparam;
       } else {
         tb.lh[tb.ix] = isLHS;
+        if (!strcmp(v, "rxdummyLhs")) {
+          tb.dummyLhs = 1;
+        }
       }
       tb.ixL=-1;
     }
@@ -299,6 +305,9 @@ static inline int handleRemainingAssignmentsCalcPropIni(nodeInfo ni, char *name,
         /* Rprintf("Duplicate %s; %d %d\n", v, tb.lh[tb.ix], tb.ini0[tb.ix]); */
         if (tb.lh[tb.ix] != isLHS){
           tb.lh[tb.ix] = isLHS;
+          if (!strcmp(v, "rxdummyLhs")) {
+            tb.dummyLhs = 1;
+          }
           if (nodeHas(ini0) && tb.ini0[tb.ix] == 1){
             sPrint(&_gbuf,"cannot have conditional initial conditions for '%s'",v);
             updateSyntaxCol();

--- a/src/parseDdt.h
+++ b/src/parseDdt.h
@@ -40,7 +40,7 @@ static inline int new_de(const char *s, int fromWhere) {
 
 static inline int isCmtLhsStatement(nodeInfo ni, char *name, char *v) {
   int hasLhs = 0;
-  if (nodeHas(cmt_statement)){
+  if (nodeHas(cmt_statement)) {
     new_or_ith(v);
     if (tb.lh[tb.ix] || tb.ini[tb.ix]){
       hasLhs=1;

--- a/src/parseDdt.h
+++ b/src/parseDdt.h
@@ -73,7 +73,10 @@ static inline int add_deCmtProp(nodeInfo ni, char *name, char *v, int hasLhs, in
 }
 
 static inline int add_deState(nodeInfo ni, char *name, char *v, int hasLhs, int fromWhere) {
-  new_or_ith(v);
+  if (new_or_ith(v)) {
+    addSymbolStr(v);
+    new_or_ith(v);
+  }
   if (((tb.ini[tb.ix] == 1 && tb.ini0[tb.ix] == 0) ||
        (tb.lh[tb.ix] == isLHS || tb.lh[tb.ix] == isLHSparam))){
     updateSyntaxCol();

--- a/src/parseFunsDosing.h
+++ b/src/parseFunsDosing.h
@@ -57,23 +57,37 @@ static inline int handleFunctionTadSingleStateCcode(transFunctions *tf,char *v2)
 	sAppend(&sb, "_%s1(", tf->v);
 	sAppend(&sbDt, "_%s1(", tf->v);
 	if (new_de(v2, 0)){
-		if (!strcmp("depot", v2)){
-			tb.hasDepot = 1;
-			aAppendN("_DEPOT_)", 8);
-		} else if (!strcmp("central", v2)){
-			tb.hasCentral = 1;
-			aAppendN("_CENTRAL_)", 10);
-		} else {
-			tb.statei++;
-			sAppend(&sb, "%d)", tb.de.n);
-			sAppend(&sbDt, "%d)", tb.de.n);
-		}
+		// cannot be lhs statements in tad style assignments
+		// also cannot be from anywhere
+		add_de(tf->ni, tf->name, v2, 0, 0);
+		aProp(tb.de.n);
 	} else {
 		new_or_ith(v2);
-		sAppend(&sb, "%d)", tb.id);
-		sAppend(&sbDt, "%d)", tb.id);
+    aProp(tb.ix);
 	}
-	// tad(cmt)
+  sAppend(&sb, "__DDT%d__)", tb.id);
+  sAppend(&sbDt, "__DDT%d__)", tb.id);
+	if (tf->isTad && (tb.dprop[tb.id] & propTad) == 0) {
+    tb.dprop[tb.id] += propTad;
+  } else if (tf->isTad0 && (tb.dprop[tb.id] & propTad0) == 0) {
+    tb.dprop[tb.id] += propTad0;
+  } else if (tf->isTafd && (tb.dprop[tb.id] & propTafd) == 0) {
+		tb.dprop[tb.id] += propTafd;
+	} else if (tf->isTafd0 && (tb.dprop[tb.id] & propTafd0) == 0) {
+		tb.dprop[tb.id] += propTafd0;
+	} else if (tf->isTlast && (tb.dprop[tb.id] & propTlast) == 0) {
+		tb.dprop[tb.id] += propTlast;
+	} else if (tf->isTlast0 && (tb.dprop[tb.id] & propTlast0) == 0) {
+		tb.dprop[tb.id] += propTlast0;
+	} else if (tf->isTfirst && (tb.dprop[tb.id] & propTfirst) == 0) {
+		tb.dprop[tb.id] += propTfirst;
+	} else if (tf->isTfirst0 && (tb.dprop[tb.id] & propTfirst0) == 0) {
+		tb.dprop[tb.id] += propTfirst0;
+	} else if (tf->isDose && (tb.dprop[tb.id] & propDose) == 0) {
+		tb.dprop[tb.id] += propDose;
+	} else if (tf->isPodo && (tb.dprop[tb.id] & propPodo) == 0) {
+		tb.dprop[tb.id] += propPodo;
+	}
 	return 1;
 }
 

--- a/src/parseFunsDosing.h
+++ b/src/parseFunsDosing.h
@@ -60,9 +60,16 @@ static inline int handleFunctionTadSingleStateCcode(transFunctions *tf,char *v2)
 		// cannot be lhs statements in tad style assignments
 		// also cannot be from anywhere
 		// temporarily turn off that this is a function
+		// This is not a function
 		int fn = tb.fn;
 		tb.fn = 0;
-		add_de(tf->ni, tf->name, v2, 0, 0);
+		// this is NOT an assignment
+		nodeInfo ni2;
+    niReset(&ni2);
+    char *name = tf->v;
+		ni2.assignment = 0;
+		ni2.ini = 0;
+		add_de(ni2, name, v2, 0, 0);
 		// turn back on that this is a function
 		tb.fn = fn;
 	} else {

--- a/src/parseFunsDosing.h
+++ b/src/parseFunsDosing.h
@@ -57,21 +57,25 @@ static inline int handleFunctionTadSingleStateCcode(transFunctions *tf,char *v2)
 	sAppend(&sb, "_%s1(", tf->v);
 	sAppend(&sbDt, "_%s1(", tf->v);
 	if (new_de(v2, 0)){
-		// cannot be lhs statements in tad style assignments
-		// also cannot be from anywhere
-		// temporarily turn off that this is a function
-		// This is not a function
-		int fn = tb.fn;
-		tb.fn = 0;
-		// this is NOT an assignment
-		nodeInfo ni2;
-    niReset(&ni2);
-    char *name = tf->v;
-		ni2.assignment = 0;
-		ni2.ini = 0;
-		add_de(ni2, name, v2, 0, 0);
-		// turn back on that this is a function
-		tb.fn = fn;
+		if (!strcmp("depot", v2)) {
+      tb.hasDepot = 1;
+      aAppendN("_DEPOT_)", 8);
+			return 1;
+		} else if (!strcmp("central", v2)) {
+			tb.hasCentral = 1;
+			aAppendN("_CENTRAL_)", 10);
+			return 1;
+		} else {
+      // cannot be lhs statements in tad style assignments
+      // also cannot be from anywhere
+      // temporarily turn off that this is a function
+      // This is not a function
+      int fn = tb.fn;
+      tb.fn = 0;
+      add_de(tf->ni, tf->name, v2, 0, 0);
+      // turn back on that this is a function
+      tb.fn = fn;
+		}
 	} else {
 		new_or_ith(v2);
 	}

--- a/src/parseFunsDosing.h
+++ b/src/parseFunsDosing.h
@@ -57,12 +57,12 @@ static inline int handleFunctionTadSingleStateCcode(transFunctions *tf,char *v2)
 	sAppend(&sb, "_%s1(", tf->v);
 	sAppend(&sbDt, "_%s1(", tf->v);
 	if (new_de(v2, 0)){
-		if (!strcmp("depot", v2)) {
-      tb.hasDepot = 1;
+		if (tb.linCmt && !strcmp("depot", v2)) {
+      tb.hasDepotCmt = 1;
       aAppendN("_DEPOT_)", 8);
 			return 1;
-		} else if (!strcmp("central", v2)) {
-			tb.hasCentral = 1;
+		} else if (tb.linCmt && !strcmp("central", v2)) {
+			tb.hasCentralCmt = 1;
 			aAppendN("_CENTRAL_)", 10);
 			return 1;
 		} else {

--- a/src/parseFunsDosing.h
+++ b/src/parseFunsDosing.h
@@ -59,11 +59,14 @@ static inline int handleFunctionTadSingleStateCcode(transFunctions *tf,char *v2)
 	if (new_de(v2, 0)){
 		// cannot be lhs statements in tad style assignments
 		// also cannot be from anywhere
+		// temporarily turn off that this is a function
+		int fn = tb.fn;
+		tb.fn = 0;
 		add_de(tf->ni, tf->name, v2, 0, 0);
-		aProp(tb.de.n);
+		// turn back on that this is a function
+		tb.fn = fn;
 	} else {
 		new_or_ith(v2);
-    aProp(tb.ix);
 	}
   sAppend(&sb, "__DDT%d__)", tb.id);
   sAppend(&sbDt, "__DDT%d__)", tb.id);

--- a/src/parseVars.h
+++ b/src/parseVars.h
@@ -115,6 +115,7 @@ static inline int skipReservedVariables(const char *s) {
 /* new symbol? if no, find it's ith */
 static inline int new_or_ith(const char *s) {
   int i;
+  tb.ix=-2;
   if (tb.fn) {tb.ix=-2; return 0;}
   if (!strcmp("lhs", s)){tb.ix=-1; return 0;}
   if (assertForbiddenVariables(s) == 0) return 0;

--- a/src/tran.c
+++ b/src/tran.c
@@ -405,8 +405,6 @@ void reset(void) {
   tb.maxtheta   = 0;
   tb.hasCmt     = 0;
   tb.maxeta     = 0;
-  tb.hasDepot   = 0;
-  tb.hasCentral = 0;
   tb.hasDepotCmt = 0;
   tb.hasCentralCmt = 0;
   tb.hasKa      = 0;

--- a/src/tran.c
+++ b/src/tran.c
@@ -422,6 +422,7 @@ void reset(void) {
   tb.linExtra   = false;
   tb.nwhile     = 0;
   tb.lvlStr     = 0;
+  tb.dummyLhs   = 0;
   tb.nInd       = 0;
   tb.simflg     = 0;
   tb.nLlik      = 0;

--- a/src/tran.h
+++ b/src/tran.h
@@ -84,8 +84,6 @@ lhs symbols?
   int maxtheta;
   int hasCmt;
   int maxeta;
-  int hasDepot;
-  int hasCentral;
   int hasDepotCmt;
   int hasCentralCmt;
   int hasKa;

--- a/src/tran.h
+++ b/src/tran.h
@@ -107,6 +107,7 @@ lhs symbols?
   int lastDdt;
   int nLlik;
   int lvlStr;
+  int dummyLhs;
 } symtab;
 
 extern symtab tb;

--- a/src/tran.h
+++ b/src/tran.h
@@ -369,10 +369,25 @@ void _rxode2parse_unprotect(void);
 
 char *getLine (char *src, int line, int *lloc);
 
-#define prop0    1
-#define propF    2
-#define propAlag 4
-#define propRate 8
-#define propDur  16
+#define prop0     1
+#define propF     2
+#define propAlag  4
+#define propRate  8
+#define propDur   16
+// tad
+#define propTad   32
+#define propTad0  64
+// tafd
+#define propTafd  128
+#define propTafd0 256
+// tlast
+#define propTlast 512
+#define propTlast0 1024
+// tfirst
+#define propTfirst 2048
+#define propTfirst0 4096
+// podo and dose
+#define propPodo 8192
+#define propDose 16384
 
 #endif // __TRAN_H__

--- a/tests/testthat/test-tad.R
+++ b/tests/testthat/test-tad.R
@@ -519,4 +519,36 @@ rxTest({
     expect_false(isTRUE(all.equal(x$tad, x$tade)))
 
   })
+
+  test_that("tad parsing", {
+
+    mod2 <- function() {
+      ini({
+        ## Table 3 from Savic 2007
+        cl  <- 17.2 # (L/hr)
+        vc  <- 45.1 # L
+        ka  <- 0.38 # 1/hr
+        mtt <- 1.37 # hr
+        f2 <-0.5    # Fraction of 1st Order portion
+        n   <- 20.1
+      })
+      model({
+        k           <- cl/vc
+        bio <- 1-f2
+        ktr = (n+1)/mtt
+        ## note that lgammafn is the same as lgamma in R.
+        d/dt(depot1) = exp(log(bio*podo(depot))+
+                             log(ktr)+n*log(ktr*tad(depot))-
+                             ktr*tad(depot)-lgammafn(n+1))-ka*depot1
+        d/dt(depot2) <- -ka*depot2
+        f(depot2) <-f2
+        d/dt(cen)   <- ka*depot1 + ka*depot2-k*cen
+      })
+    }
+
+    mod2 <- mod2()
+
+    mod2$simulationModel
+
+  })
 })

--- a/tests/testthat/test-tad.R
+++ b/tests/testthat/test-tad.R
@@ -546,9 +546,33 @@ rxTest({
       })
     }
 
-    mod2 <- mod2()
+    expect_error(mod2())
 
-    mod2$simulationModel
+    mod2 <- function() {
+      ini({
+        ## Table 3 from Savic 2007
+        cl  <- 17.2 # (L/hr)
+        vc  <- 45.1 # L
+        ka  <- 0.38 # 1/hr
+        mtt <- 1.37 # hr
+        f2 <-0.5    # Fraction of 1st Order portion
+        n   <- 20.1
+      })
+      model({
+        k           <- cl/vc
+        bio <- 1-f2
+        ktr = (n+1)/mtt
+        ## note that lgammafn is the same as lgamma in R.
+        d/dt(depot1) = exp(log(bio*podo(depot1))+
+                             log(ktr)+n*log(ktr*tad(depot1))-
+                             ktr*tad(depot1)-lgammafn(n+1))-ka*depot1
+        d/dt(depot2) <- -ka*depot2
+        f(depot2) <-f2
+        d/dt(cen)   <- ka*depot1 + ka*depot2-k*cen
+      })
+    }
+
+    mod2 <- mod2()
 
   })
 })

